kv/kvserver: TestReplicateQueueDownReplicate failed

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1913442&tab=buildLog) on [master@d8b16b51416d1874a3d80801560a574c3807e291](https://github.com/cockroachdb/cockroach/commits/d8b16b51416d1874a3d80801560a574c3807e291):

```
W200501 16:56:04.059715 416114 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200501 16:56:04.049731 893063 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36739: context canceled
W200501 16:56:04.081382 414732 ts/db.go:194  [n2,ts-poll] error writing time series data: node unavailable; try another peer
I200501 16:56:04.095070 893005 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42207: context canceled
W200501 16:56:04.096266 474170 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200501 16:56:04.100973 893004 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:41257: context canceled
I200501 16:56:04.104355 893003 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:41307: context canceled
W200501 16:56:04.143288 412272 ts/db.go:194  [n1,ts-poll] error writing time series data: node unavailable; try another peer
E200501 16:56:04.148580 893108 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200501 16:56:04.159898 484714 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 2: EOF:
W200501 16:56:04.161491 464171 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200501 16:56:04.162430 463354 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 2: EOF:
I200501 16:56:04.166049 893107 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:41257: context canceled
W200501 16:56:04.189121 473369 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200501 16:56:04.190290 415522 server/node.go:722  [n3,summaries] error recording status summaries: node unavailable; try another peer
W200501 16:56:04.193285 473167 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200501 16:56:04.196152 415151 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200501 16:56:04.198365 882957 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200501 16:56:04.198965 474171 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
W200501 16:56:04.199009 463349 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200501 16:56:04.199292 891054 rpc/nodedialer/nodedialer.go:160  [n3,replicaGC,s3,r33/7:{m-/Table/Syste…}] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42207: context canceled
I200501 16:56:04.201336 891054 rpc/nodedialer/nodedialer.go:160  [n3,replicaGC,s3,r33/7:{m-/Table/Syste…}] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36739: context canceled
I200501 16:56:04.203528 891054 rpc/nodedialer/nodedialer.go:160  [n3,replicaGC,s3,r33/7:{m-/Table/Syste…}] unable to connect to n4: failed to connect to n4 at 127.0.0.1:41307: context canceled
E200501 16:56:04.205603 891054 kv/kvserver/queue.go:1092  [n3,replicaGC,s3,r33/7:{m-/Table/Syste…}] failed to send RPC: sending to all 5 replicas failed; last error: <nil> failed to connect to n4 at 127.0.0.1:41307: context canceled
E200501 16:56:04.212689 893009 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200501 16:56:04.218419 419530 ts/db.go:194  [n5,ts-poll] error writing time series data: failed to send RPC: sending to all 3 replicas failed; last error: (err: node unavailable; try another peer) <nil>
W200501 16:56:04.252381 415464 kv/kvserver/store.go:1627  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200501 16:56:04.279081 892898 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42915: context canceled
E200501 16:56:04.280376 893123 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200501 16:56:04.314638 415464 kv/kvserver/store.go:1627  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200501 16:56:04.329501 415449 ts/db.go:197  [n3,ts-poll] node unavailable; try another peer
E200501 16:56:04.381421 893208 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200501 16:56:04.382235 893133 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestReplicateQueueDownReplicate (334.66s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n2,s2):4, (n3,s3):7LEARNER, next=8, gen=18, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n2,s2):4, next=8, gen=19, sticky=9223372036.854775807,2147483647]
        goroutine 412188 [running]:
        runtime/debug.Stack(0xc008b7ba00, 0x72138c0, 0xc0003d3b80)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x732dac0, 0xc002758400, 0xc008b7ba00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc002758400)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc002758400, 0x63683a0)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1932178&tab=buildLog) on [master@c1abb272c94a437f1df9175fc30dc6a6729d3338](https://github.com/cockroachdb/cockroach/commits/c1abb272c94a437f1df9175fc30dc6a6729d3338):

```
E200509 22:07:47.368693 1032827 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200509 22:07:47.372174 444954 kv/kvserver/store.go:1627  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200509 22:07:47.408775 440795 kv/kvserver/store.go:1627  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200509 22:07:47.417486 443021 kv/kvserver/store.go:1627  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200509 22:07:47.457960 440795 kv/kvserver/store.go:1627  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200509 22:07:47.462610 1032884 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200509 22:07:47.463497 440753 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 133 iterations on up to 7903 requests
W200509 22:07:47.474115 443021 kv/kvserver/store.go:1627  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200509 22:07:47.479688 443567 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200509 22:07:47.482088 447150 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200509 22:07:47.482624 447150 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200509 22:07:47.483054 447150 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200509 22:07:47.501629 443010 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200509 22:07:47.502979 443010 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (replicate): node unavailable; try another peer
E200509 22:07:47.512155 1032885 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200509 22:07:47.513306 1032869 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200509 22:07:47.522381 1032829 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200509 22:07:47.540549 1032888 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200509 22:07:47.560042 443567 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200509 22:07:47.560878 443567 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (replicate): node unavailable; try another peer
E200509 22:07:47.565395 1032922 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200509 22:07:47.578352 447447 server/node.go:724  [n5,summaries] error recording status summaries: node unavailable; try another peer
W200509 22:07:47.598761 444954 kv/kvserver/store.go:1627  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200509 22:07:47.722632 1032896 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200509 22:07:47.724542 1032833 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200509 22:07:47.727245 1032834 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200509 22:07:47.728048 1032925 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200509 22:07:47.729458 1032926 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200509 22:07:47.915863 1032786 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n4] [JOB 208] compacted L0 [000292 000294] (77 K) + L6 [000293] (360 K) -> L6 [000295] (357 K), in 0.6s, output rate 570 K/s
I200509 22:07:47.917075 1032786 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 208] sstable deleted 000292
I200509 22:07:47.917448 1032786 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 208] sstable deleted 000293
I200509 22:07:47.917819 1032786 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 208] sstable deleted 000294
E200509 22:07:47.925634 1032934 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestReplicateQueueDownReplicate (376.52s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):4, (n3,s3):5, (n4,s4):13LEARNER, next=14, gen=30, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):4, (n3,s3):5, next=14, gen=31, sticky=9223372036.854775807,2147483647]
        goroutine 440637 [running]:
        runtime/debug.Stack(0xc02c599a00, 0x7250300, 0xc0119eb1e0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7371b60, 0xc0102bee00, 0xc02c599a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc0102bee00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc0102bee00, 0x6380e20)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1932308&tab=buildLog) on [master@c1abb272c94a437f1df9175fc30dc6a6729d3338](https://github.com/cockroachdb/cockroach/commits/c1abb272c94a437f1df9175fc30dc6a6729d3338):

```
W200510 03:07:28.817726 447138 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200510 03:07:28.819274 1319623 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:45911: context canceled
W200510 03:07:28.826119 508640 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200510 03:07:28.828101 444920 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200510 03:07:28.832751 526396 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 5: EOF:
W200510 03:07:28.838255 525552 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200510 03:07:28.841660 1319596 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45539: context canceled
I200510 03:07:28.844083 1319597 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:34773: context canceled
I200510 03:07:28.845012 1319631 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:45911: context canceled
W200510 03:07:28.845475 440753 kv/kvserver/store.go:1627  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200510 03:07:28.849709 1133566 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200510 03:07:28.852527 1319630 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:41693: context canceled
W200510 03:07:28.866186 444222 kv/kvserver/store_raft.go:509  [n4,s4,r2/5:/System/NodeLiveness{-Max}] handle raft ready: 1.1s [applied=6, batches=3, state_assertions=1]
I200510 03:07:28.878848 1319669 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45539: context canceled
W200510 03:07:28.884281 445734 kv/kvserver/store.go:1627  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200510 03:07:28.896632 443532 kv/kvserver/store.go:1627  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200510 03:07:28.930112 447211 kv/kvserver/replica_destroy.go:217  [n4,s4,r33/10:{m-/Table/Syste…}] removed 5 (0+5) keys in 669ms [clear=669ms commit=1ms]
W200510 03:07:28.931442 447211 kv/kvserver/store.go:1816  [n4,s4,r33/10:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
W200510 03:07:28.934935 447209 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: EOF:
W200510 03:07:28.953656 443532 kv/kvserver/store.go:1627  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200510 03:07:28.980741 444328 ts/db.go:194  [n4,ts-poll] error writing time series data: node unavailable; try another peer
W200510 03:07:29.068571 443532 kv/kvserver/store.go:1627  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200510 03:07:29.092436 445734 kv/kvserver/store.go:1627  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200510 03:07:29.244775 1319696 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200510 03:07:29.274595 1319817 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200510 03:07:29.436126 1319792 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200510 03:07:29.485744 445734 kv/kvserver/store.go:1627  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200510 03:07:29.512249 1304233 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n4] [JOB 189] compacted L0 [000254 000255 000261 000262] (218 K) + L6 [000248] (472 K) -> L6 [000263] (497 K), in 15.2s, output rate 33 K/s
I200510 03:07:29.513710 1304233 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 189] sstable deleted 000248
I200510 03:07:29.514170 1304233 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 189] sstable deleted 000254
I200510 03:07:29.514577 1304233 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 189] sstable deleted 000255
I200510 03:07:29.514955 1304233 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 189] sstable deleted 000261
I200510 03:07:29.515308 1304233 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 189] sstable deleted 000262
--- FAIL: TestReplicateQueueDownReplicate (608.88s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n3,s3):3, (n4,s4):10LEARNER, next=11, gen=24, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=11, gen=25, sticky=9223372036.854775807,2147483647]
        goroutine 440578 [running]:
        runtime/debug.Stack(0xc03b483a00, 0x7250300, 0xc037a0d720)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7371b60, 0xc018308e00, 0xc03b483a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc018308e00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc018308e00, 0x6380e20)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1932684&tab=buildLog) on [master@c1abb272c94a437f1df9175fc30dc6a6729d3338](https://github.com/cockroachdb/cockroach/commits/c1abb272c94a437f1df9175fc30dc6a6729d3338):

```
W200510 06:03:09.333013 434625 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200510 06:03:09.340975 427648 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
W200510 06:03:09.346030 429272 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200510 06:03:09.346074 428580 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (gc): node unavailable; try another peer
W200510 06:03:09.347059 436979 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200510 06:03:09.347722 428580 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200510 06:03:09.348381 428580 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (split): node unavailable; try another peer
I200510 06:03:09.348773 428580 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200510 06:03:09.349079 428580 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200510 06:03:09.351566 428580 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200510 06:03:09.351985 428580 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
W200510 06:03:09.352952 429177 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200510 06:03:09.353661 501486 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200510 06:03:09.354147 509729 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200510 06:03:09.354285 436560 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200510 06:03:09.382049 912012 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:34137: context canceled
I200510 06:03:09.382709 912009 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38851: context canceled
W200510 06:03:09.384800 430055 kv/kvserver/store.go:1627  [n4,s4,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200510 06:03:09.403518 432071 ts/db.go:194  [n5,ts-poll] error writing time series data: node unavailable; try another peer
I200510 06:03:09.413148 912018 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:37033: context canceled
I200510 06:03:09.415416 912068 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:34137: context canceled
I200510 06:03:09.419478 912067 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35795: context canceled
I200510 06:03:09.423931 912069 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:35373: context canceled
E200510 06:03:09.428798 912084 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200510 06:03:09.430373 911954 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200510 06:03:09.492816 430055 kv/kvserver/store.go:1627  [n4,s4,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200510 06:03:09.753575 910199 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n5] [JOB 221] compacted L0 [000326 000321 000324 000322 000323] (62 K) + L6 [000320] (378 K) -> L6 [000327] (379 K), in 1.7s, output rate 219 K/s
I200510 06:03:09.754717 910199 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 221] sstable deleted 000320
I200510 06:03:09.755101 910199 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 221] sstable deleted 000321
I200510 06:03:09.755494 910199 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 221] sstable deleted 000322
I200510 06:03:09.755914 910199 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 221] sstable deleted 000323
I200510 06:03:09.756182 910199 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 221] sstable deleted 000324
I200510 06:03:09.756414 910199 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 221] sstable deleted 000326
--- FAIL: TestReplicateQueueDownReplicate (287.56s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n2,s2):4, (n3,s3):5, (n4,s4):3, (n5,s5):18LEARNER, next=19, gen=40, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n2,s2):4, (n3,s3):5, (n4,s4):3, next=19, gen=41, sticky=9223372036.854775807,2147483647]
        goroutine 425757 [running]:
        runtime/debug.Stack(0xc024259a00, 0x695dc40, 0xc03499c620)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x6a7f4a0, 0xc044edc300, 0xc024259a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc044edc300)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc044edc300, 0x5a8e760)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1935014&tab=buildLog) on [master@e6b47088aba9c9978501b966a0e88aeb273d9990](https://github.com/cockroachdb/cockroach/commits/e6b47088aba9c9978501b966a0e88aeb273d9990):

```
I200511 16:33:13.438514 1281296 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:34723: context canceled
I200511 16:33:13.440814 1281298 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:43843: context canceled
I200511 16:33:13.441960 442667 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (split): node unavailable; try another peer
W200511 16:33:13.448801 503242 kv/kvserver/store.go:1816  [n5,s5,r33/10:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
I200511 16:33:13.451852 1281430 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:43843: context canceled
I200511 16:33:13.458446 1281433 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:46041: context canceled
W200511 16:33:13.460633 438698 kv/kvserver/node_liveness.go:496  [n1,liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200511 16:33:13.471543 441796 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200511 16:33:13.483559 438697 ts/db.go:194  [n1,ts-poll] error writing time series data: node unavailable; try another peer
I200511 16:33:13.486170 442667 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200511 16:33:13.488016 503202 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 2: EOF:
I200511 16:33:13.492214 442667 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200511 16:33:13.492993 442667 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200511 16:33:13.493698 442667 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200511 16:33:13.494248 442667 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200511 16:33:13.494876 442667 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200511 16:33:13.505139 444832 kv/kvserver/store.go:1627  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200511 16:33:13.508943 1281282 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:44415: context canceled
I200511 16:33:13.509817 1281459 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:34723: context canceled
W200511 16:33:13.547005 441548 kv/kvserver/store.go:1627  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200511 16:33:13.565841 1281450 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:44415: context canceled
E200511 16:33:13.596509 1281352 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200511 16:33:13.621301 444832 kv/kvserver/store.go:1627  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200511 16:33:13.696857 1267053 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n5] [JOB 174] compacted L0 [000250 000251] (104 K) + L6 [000244] (460 K) -> L6 [000252] (445 K), in 13.2s, output rate 34 K/s
I200511 16:33:13.699791 1267053 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 174] sstable deleted 000244
I200511 16:33:13.701296 1267053 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 174] sstable deleted 000250
I200511 16:33:13.702690 1267053 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 174] sstable deleted 000251
I200511 16:33:13.704975 1281328 vendor/github.com/cockroachdb/pebble/compaction.go:1132  [n5] [JOB 181] compacting L0 [000258 000259] (132 K) + L6 [000252] (445 K)
I200511 16:33:13.710774 1281328 vendor/github.com/cockroachdb/pebble/compaction.go:1287  [n5] [JOB 181] compacting: sstable created 000260
I200511 16:33:14.629696 1281328 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n5] [JOB 181] compacted L0 [000258 000259] (132 K) + L6 [000252] (445 K) -> L6 [000260] (476 K), in 0.9s, output rate 516 K/s
I200511 16:33:14.631950 1281328 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 181] sstable deleted 000252
I200511 16:33:14.633114 1281328 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 181] sstable deleted 000258
I200511 16:33:14.634019 1281328 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 181] sstable deleted 000259
--- FAIL: TestReplicateQueueDownReplicate (583.04s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n4,s4):7, (n5,s5):10LEARNER, next=11, gen=27, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n4,s4):7, next=11, gen=28, sticky=9223372036.854775807,2147483647]
        goroutine 438486 [running]:
        runtime/debug.Stack(0xc04b05fa00, 0x72560c0, 0xc00ac165e0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x73778c0, 0xc044634600, 0xc04b05fa00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc044634600)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc044634600, 0x6385978)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1937152&tab=buildLog) on [master@298ef0ea16314ed5307088470f159ef2b25385c3](https://github.com/cockroachdb/cockroach/commits/298ef0ea16314ed5307088470f159ef2b25385c3):

```
E200512 16:45:11.760101 1348110 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:11.839233 1348111 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:11.899771 1348192 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200512 16:45:11.963258 1347898 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n3] [JOB 189] compacted L0 [000257 000258 000264 000265] (227 K) + L6 [000251] (446 K) -> L6 [000266] (479 K), in 1.3s, output rate 369 K/s
W200512 16:45:11.963891 454116 kv/kvserver/store.go:1631  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200512 16:45:11.965069 1347898 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 189] sstable deleted 000251
I200512 16:45:11.966037 1347898 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 189] sstable deleted 000257
I200512 16:45:11.966609 1347898 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 189] sstable deleted 000258
I200512 16:45:11.967324 1347898 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 189] sstable deleted 000264
I200512 16:45:11.968249 1347898 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 189] sstable deleted 000265
E200512 16:45:12.031830 1348248 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.034976 1348210 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.090230 1348283 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.096949 1348315 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.097875 1348250 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.099672 1348269 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.147938 1348273 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.150319 1348286 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.151795 1348046 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.152940 1348256 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.234666 1348340 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200512 16:45:12.259824 455554 kv/kvserver/store.go:1631  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200512 16:45:12.287134 1348341 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.294348 1348355 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.308044 1348373 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200512 16:45:12.340315 451276 kv/kvserver/store.go:1631  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200512 16:45:12.344096 454116 kv/kvserver/store.go:1631  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200512 16:45:12.350255 1348330 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200512 16:45:12.352605 1348331 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200512 16:45:12.379372 451327 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 114 iterations on up to 10081 requests
W200512 16:45:12.386265 453460 kv/kvserver/store.go:1631  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200512 16:45:12.511043 1348322 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200512 16:45:12.519491 457599 kv/kvserver/store.go:1631  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (638.83s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n4,s4):4, (n3,s3):9LEARNER, next=10, gen=22, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n4,s4):4, next=10, gen=23, sticky=9223372036.854775807,2147483647]
        goroutine 451101 [running]:
        runtime/debug.Stack(0xc023bcba00, 0x7261000, 0xc04073c640)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7382900, 0xc01cee7300, 0xc023bcba00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc01cee7300)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc01cee7300, 0x638f3d8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1939077&tab=buildLog) on [master@3531318daf434670bb23ae68b8e8b8a4e4801879](https://github.com/cockroachdb/cockroach/commits/3531318daf434670bb23ae68b8e8b8a4e4801879):

```
W200513 06:25:09.218424 450467 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200513 06:25:09.224990 1021995 kv/kvserver/intentresolver/intent_resolver.go:539  failed to cleanup transaction intents: failed to resolve intents: aborted during DistSender.Send: context canceled
W200513 06:25:09.230088 452110 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200513 06:25:09.231448 891665 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200513 06:25:09.232813 457941 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: EOF:
I200513 06:25:09.235992 1022149 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35637: context canceled
W200513 06:25:09.248759 596971 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200513 06:25:09.249553 1022152 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36787: context canceled
I200513 06:25:09.267180 1022334 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:33961: context canceled
I200513 06:25:09.278009 1022336 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43277: context canceled
I200513 06:25:09.282250 1022335 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35637: context canceled
W200513 06:25:09.283814 450259 ts/db.go:194  [n4,ts-poll] error writing time series data: node unavailable; try another peer
W200513 06:25:09.289622 449141 ts/db.go:194  [n3,ts-poll] error writing time series data: failed to send RPC: sending to all 1 replicas failed; last error: <nil> rpc error: code = Unavailable desc = transport is closing
I200513 06:25:09.284292 1022337 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36787: context canceled
E200513 06:25:09.302029 1022361 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200513 06:25:09.302730 1022421 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200513 06:25:09.318362 1022409 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200513 06:25:09.321531 450196 kv/kvserver/store.go:1631  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200513 06:25:09.325661 1022408 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42307: context canceled
W200513 06:25:09.374982 450196 kv/kvserver/store.go:1631  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200513 06:25:09.429899 1022373 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200513 06:25:09.467950 450196 kv/kvserver/store.go:1631  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200513 06:25:09.499545 1022447 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200513 06:25:09.506132 452259 ts/db.go:194  [n5,ts-poll] error writing time series data: node unavailable; try another peer
E200513 06:25:09.708531 1022470 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200513 06:25:09.708778 1022499 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200513 06:25:09.735069 1019914 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n4] [JOB 216] compacted L0 [000304 000299 000302 000300 000301] (68 K) + L6 [000298] (366 K) -> L6 [000305] (375 K), in 2.4s, output rate 159 K/s
I200513 06:25:09.736635 1019914 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 216] sstable deleted 000298
I200513 06:25:09.737507 1019914 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 216] sstable deleted 000299
I200513 06:25:09.737924 1019914 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 216] sstable deleted 000300
I200513 06:25:09.738471 1019914 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 216] sstable deleted 000301
I200513 06:25:09.738974 1019914 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 216] sstable deleted 000302
I200513 06:25:09.739424 1019914 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 216] sstable deleted 000304
--- FAIL: TestReplicateQueueDownReplicate (397.98s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n2,s2):4, (n4,s4):15LEARNER, next=16, gen=34, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n2,s2):4, next=16, gen=35, sticky=9223372036.854775807,2147483647]
        goroutine 446185 [running]:
        runtime/debug.Stack(0xc04fff7a00, 0x69e0fc0, 0xc02a0e7bc0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x6b04240, 0xc0496eea00, 0xc04fff7a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc0496eea00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc0496eea00, 0x5ad4fe8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1940508&tab=buildLog) on [master@845a170e82feb6bb5edb41523267d678f7706439](https://github.com/cockroachdb/cockroach/commits/845a170e82feb6bb5edb41523267d678f7706439):

```
I200513 17:29:35.145724 1289064 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:34917: context canceled
W200513 17:29:35.148285 459744 kv/kvserver/store.go:1652  [n3,s3,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200513 17:29:35.150456 1289066 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:34707: context canceled
W200513 17:29:35.153221 456370 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200513 17:29:35.201467 1288635 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:34917: context canceled
I200513 17:29:35.203466 1288636 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:34707: context canceled
I200513 17:29:35.211236 1289071 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39831: context canceled
I200513 17:29:35.216701 1289072 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:34707: context canceled
W200513 17:29:35.224589 463383 ts/db.go:194  [n5,ts-poll] error writing time series data: node unavailable; try another peer
W200513 17:29:35.225355 463383 ts/db.go:197  [n5,ts-poll] node unavailable; try another peer
I200513 17:29:35.245098 463396 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200513 17:29:35.246454 463396 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200513 17:29:35.254985 459744 kv/kvserver/store.go:1652  [n3,s3,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200513 17:29:35.256257 456370 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200513 17:29:35.336233 1289113 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200513 17:29:35.336673 1289092 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200513 17:29:35.350454 1289157 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200513 17:29:35.403184 461140 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200513 17:29:35.404292 461140 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicate): node unavailable; try another peer
E200513 17:29:35.420677 1289132 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200513 17:29:35.458278 459744 kv/kvserver/store.go:1652  [n3,s3,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200513 17:29:35.486916 456370 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200513 17:29:35.526605 458987 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (merge): node unavailable; try another peer
E200513 17:29:35.553705 1289120 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200513 17:29:35.574415 458987 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200513 17:29:35.575644 458987 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (replicate): node unavailable; try another peer
E200513 17:29:35.588619 1289121 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200513 17:29:35.605211 1289222 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200513 17:29:35.622525 458970 ts/db.go:194  [n2,ts-poll] error writing time series data: result is ambiguous (server shutdown)
W200513 17:29:35.622977 458970 ts/db.go:197  [n2,ts-poll] node unavailable; try another peer
E200513 17:29:35.629461 1289209 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200513 17:29:35.630610 1289210 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200513 17:29:35.692372 456226 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 130 iterations on up to 9869 requests
--- FAIL: TestReplicateQueueDownReplicate (572.70s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n3,s3):11LEARNER, next=12, gen=26, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n4,s4):3, next=12, gen=27, sticky=9223372036.854775807,2147483647]
        goroutine 456138 [running]:
        runtime/debug.Stack(0xc062aeda00, 0x72d4ea0, 0xc03167ad40)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x73f8120, 0xc00c986500, 0xc062aeda00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc00c986500)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc00c986500, 0x63c8ae8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1944621&tab=buildLog) on [master@d1d0d17e31da1d3ee88276e94953c48bf197fca2](https://github.com/cockroachdb/cockroach/commits/d1d0d17e31da1d3ee88276e94953c48bf197fca2):

```
E200515 03:38:30.910406 1096322 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 03:38:30.921440 1096517 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 03:38:30.929656 1096518 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 03:38:31.059790 1096534 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 03:38:31.089666 1096491 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200515 03:38:31.095424 453380 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
W200515 03:38:31.098410 453492 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200515 03:38:31.109874 1096548 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 03:38:31.111345 1096550 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 03:38:31.121161 1096564 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 03:38:31.127812 1096568 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200515 03:38:31.140097 460548 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200515 03:38:31.187873 456860 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200515 03:38:31.198083 460548 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200515 03:38:31.260253 1096569 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 03:38:31.261148 1096537 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200515 03:38:31.261519 453525 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 107 iterations on up to 8166 requests
E200515 03:38:31.274632 1096584 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200515 03:38:31.301645 460548 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200515 03:38:31.492352 1096612 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200515 03:38:31.527478 460548 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200515 03:38:31.528927 458431 ts/db.go:197  [n4,ts-poll] node unavailable; try another peer
W200515 03:38:31.608923 458416 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200515 03:38:31.659093 458416 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200515 03:38:31.688941 1096614 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 03:38:31.690856 1096617 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200515 03:38:31.702258 1095459 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n3] [JOB 196] compacted L0 [000289 000290 000296 000297] (186 K) + L6 [000283] (370 K) -> L6 [000298] (398 K), in 2.7s, output rate 146 K/s
I200515 03:38:31.703605 1095459 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 196] sstable deleted 000283
I200515 03:38:31.704016 1095459 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 196] sstable deleted 000289
I200515 03:38:31.704554 1095459 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 196] sstable deleted 000290
I200515 03:38:31.705032 1095459 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 196] sstable deleted 000296
I200515 03:38:31.705441 1095459 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n3] [JOB 196] sstable deleted 000297
E200515 03:38:31.716345 1096597 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestReplicateQueueDownReplicate (397.75s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n4,s4):4, (n3,s3):14LEARNER, next=15, gen=32, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n4,s4):4, next=15, gen=33, sticky=9223372036.854775807,2147483647]
        goroutine 453227 [running]:
        runtime/debug.Stack(0xc01fa8ba00, 0x7307f20, 0xc029fa64c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x742bba0, 0xc0141e1500, 0xc01fa8ba00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc0141e1500)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc0141e1500, 0x63eb8d0)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1945077&tab=buildLog) on [master@1bd006c3a9f90cc8072d941089a5ebecff051a3e](https://github.com/cockroachdb/cockroach/commits/1bd006c3a9f90cc8072d941089a5ebecff051a3e):

```
I200515 16:22:29.472474 1301411 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 213] sstable deleted 000303
I200515 16:22:29.473146 1301411 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 213] sstable deleted 000304
I200515 16:22:29.474196 1301411 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 213] sstable deleted 000305
I200515 16:22:29.475210 1301411 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 213] sstable deleted 000306
I200515 16:22:29.475705 1301411 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 213] sstable deleted 000307
I200515 16:22:29.476229 1301411 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 213] sstable deleted 000309
W200515 16:22:29.503194 461948 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200515 16:22:29.556384 460840 ts/db.go:197  [n3,ts-poll] node unavailable; try another peer
W200515 16:22:29.562589 464335 kv/kvserver/store.go:1652  [n5,s5,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200515 16:22:29.570923 1302254 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200515 16:22:29.572753 460115 kv/kvserver/store.go:1652  [n2,s2,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200515 16:22:29.574365 1302255 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200515 16:22:29.627094 460115 kv/kvserver/store.go:1652  [n2,s2,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200515 16:22:29.629714 1302257 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 16:22:29.632477 1302351 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 16:22:29.653606 1302352 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200515 16:22:29.653981 457651 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
E200515 16:22:29.663257 1302258 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200515 16:22:29.697989 461948 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200515 16:22:29.705922 1302398 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200515 16:22:29.739705 460115 kv/kvserver/store.go:1652  [n2,s2,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200515 16:22:29.777447 1302420 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 16:22:29.777855 1302317 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 16:22:29.779027 1302440 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 16:22:29.780013 1302441 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200515 16:22:29.796974 460839 server/status/runtime.go:499  [n3] runtime stats: 12 GiB RSS, 654 goroutines, 1.5 GiB/224 MiB/1.8 GiB GO alloc/idle/total(stale), 209 MiB/302 MiB CGO alloc/total, 4937.7 CGO/sec, 342.9/31.4 %(u/s)time, 0.0 %gc (1x), 1.9 MiB/1.9 MiB (r/w)net
E200515 16:22:29.826112 1302368 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 16:22:29.832382 1302399 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200515 16:22:29.847616 457659 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 158 iterations on up to 9890 requests
E200515 16:22:29.970987 1302469 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 16:22:29.971850 1302485 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 16:22:29.972471 1302486 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200515 16:22:29.973311 1302422 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestReplicateQueueDownReplicate (595.19s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):4, (n2,s2):3, (n5,s5):15LEARNER, next=16, gen=34, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):4, (n2,s2):3, next=16, gen=35, sticky=9223372036.854775807,2147483647]
        goroutine 457611 [running]:
        runtime/debug.Stack(0xc05c425a00, 0x6a172e0, 0xc044a96a00)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x6b3afa0, 0xc049941600, 0xc05c425a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc049941600)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc049941600, 0x5afa440)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1951972&tab=buildLog) on [master@7ab7fb86f0634df1f5a0b04460e1f3a1d6bead1f](https://github.com/cockroachdb/cockroach/commits/7ab7fb86f0634df1f5a0b04460e1f3a1d6bead1f):

```
W200518 19:52:41.000452 454269 kv/kvserver/store.go:1652  [n4,s4,r2/4:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
I200518 19:52:41.034201 1088908 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:33071: context canceled
I200518 19:52:41.039031 1088907 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:44871: context canceled
E200518 19:52:41.062080 1089029 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200518 19:52:41.073031 1088976 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200518 19:52:41.099157 454268 kv/kvserver/store.go:1652  [n4,s4,r6/5:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200518 19:52:41.111217 454267 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200518 19:52:41.114531 454269 kv/kvserver/store.go:1652  [n4,s4,r2/4:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
E200518 19:52:41.136244 1088978 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200518 19:52:41.174383 1088408 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n4] [JOB 229] compacted L0 [000321 000316 000319 000317 000318] (54 K) + L6 [000315] (368 K) -> L6 [000322] (386 K), in 1.0s, output rate 402 K/s
I200518 19:52:41.175967 1088408 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 229] sstable deleted 000315
I200518 19:52:41.176725 1088408 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 229] sstable deleted 000316
I200518 19:52:41.177454 1088408 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 229] sstable deleted 000317
I200518 19:52:41.178407 1088408 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 229] sstable deleted 000318
I200518 19:52:41.179357 1088408 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 229] sstable deleted 000319
I200518 19:52:41.180064 1088408 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n4] [JOB 229] sstable deleted 000321
E200518 19:52:41.182301 1088993 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200518 19:52:41.184280 1089075 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200518 19:52:41.203953 454670 server/node.go:718  [n4,summaries] error recording status summaries: node unavailable; try another peer
E200518 19:52:41.230703 1089046 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200518 19:52:41.232214 1089047 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200518 19:52:41.255858 450069 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
E200518 19:52:41.271035 1089013 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200518 19:52:41.282893 1089014 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200518 19:52:41.297292 454269 kv/kvserver/store.go:1652  [n4,s4,r2/4:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200518 19:52:41.328024 454268 kv/kvserver/store.go:1652  [n4,s4,r6/5:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200518 19:52:41.331302 454267 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200518 19:52:41.384401 1089070 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200518 19:52:41.413484 1089071 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200518 19:52:41.445249 450196 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 146 iterations on up to 7941 requests
I200518 19:52:41.496586 454311 server/status/runtime.go:499  [n4] runtime stats: 11 GiB RSS, 451 goroutines, 1.3 GiB/253 MiB/1.6 GiB GO alloc/idle/total(stale), 127 MiB/199 MiB CGO alloc/total, 5373.2 CGO/sec, 355.2/30.5 %(u/s)time, 0.2 %gc (1x), 1.1 MiB/1.1 MiB (r/w)net
E200518 19:52:41.533686 1089079 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200518 19:52:41.536365 1089127 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestReplicateQueueDownReplicate (364.83s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):4, (n2,s2):5, (n4,s4):18LEARNER, next=19, gen=40, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):4, (n2,s2):5, next=19, gen=41, sticky=9223372036.854775807,2147483647]
        goroutine 449960 [running]:
        runtime/debug.Stack(0xc047685a00, 0x732c240, 0xc021e84e00)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7450280, 0xc01b4ad100, 0xc047685a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc01b4ad100)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc01b4ad100, 0x63ffdf0)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1952578&tab=buildLog) on [master@7ab7fb86f0634df1f5a0b04460e1f3a1d6bead1f](https://github.com/cockroachdb/cockroach/commits/7ab7fb86f0634df1f5a0b04460e1f3a1d6bead1f):

```
E200519 00:10:11.704447 1207759 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200519 00:10:11.812362 1207778 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200519 00:10:11.816072 461384 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200519 00:10:11.866977 1207815 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200519 00:10:11.934159 458454 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200519 00:10:11.962251 1207809 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200519 00:10:12.073506 1207944 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200519 00:10:12.077278 1207909 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200519 00:10:12.092827 454559 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200519 00:10:12.146090 456766 kv/kvserver/store.go:1652  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200519 00:10:12.152211 454559 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200519 00:10:12.158323 461384 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200519 00:10:12.201515 456766 kv/kvserver/store.go:1652  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200519 00:10:12.242029 454559 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200519 00:10:12.245387 1207880 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200519 00:10:12.269156 1207913 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200519 00:10:12.270248 1207914 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200519 00:10:12.285674 458454 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200519 00:10:12.294390 456766 kv/kvserver/store.go:1652  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200519 00:10:12.324565 1193318 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n5] [JOB 200] compacted L0 [000297 000292 000295 000293 000294] (85 K) + L6 [000291] (405 K) -> L6 [000298] (400 K), in 10.5s, output rate 38 K/s
I200519 00:10:12.325998 1193318 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 200] sstable deleted 000291
I200519 00:10:12.326589 1193318 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 200] sstable deleted 000292
I200519 00:10:12.327004 1193318 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 200] sstable deleted 000293
I200519 00:10:12.327601 1193318 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 200] sstable deleted 000294
I200519 00:10:12.328431 1193318 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 200] sstable deleted 000295
I200519 00:10:12.328972 1193318 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 200] sstable deleted 000297
I200519 00:10:12.329846 1207816 vendor/github.com/cockroachdb/pebble/compaction.go:1132  [n5] [JOB 206] compacting L0 [000304 000305] (68 K) + L6 [000298] (400 K)
I200519 00:10:12.334705 1207816 vendor/github.com/cockroachdb/pebble/compaction.go:1287  [n5] [JOB 206] compacting: sstable created 000306
E200519 00:10:12.380597 1207985 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200519 00:10:12.778244 1207816 vendor/github.com/cockroachdb/pebble/compaction.go:1170  [n5] [JOB 206] compacted L0 [000304 000305] (68 K) + L6 [000298] (400 K) -> L6 [000306] (398 K), in 0.4s, output rate 890 K/s
I200519 00:10:12.779884 1207816 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 206] sstable deleted 000298
I200519 00:10:12.780571 1207816 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 206] sstable deleted 000304
I200519 00:10:12.781153 1207816 vendor/github.com/cockroachdb/pebble/compaction.go:1858  [n5] [JOB 206] sstable deleted 000305
--- FAIL: TestReplicateQueueDownReplicate (448.70s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):5, (n3,s3):3, (n5,s5):15LEARNER, next=16, gen=34, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):5, (n3,s3):3, next=16, gen=35, sticky=9223372036.854775807,2147483647]
        goroutine 454348 [running]:
        runtime/debug.Stack(0xc040275a00, 0x732c240, 0xc0313cf500)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7450280, 0xc035cb4700, 0xc040275a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc035cb4700)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc035cb4700, 0x63ffdf0)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1954189&tab=buildLog) on [master@73631089c58554c289d0fb1b26cad62667388335](https://github.com/cockroachdb/cockroach/commits/73631089c58554c289d0fb1b26cad62667388335):

```
W200519 14:07:10.945496 453648 kv/txn.go:607  [n1,replication-reporter] failure aborting transaction: node unavailable; try another peer; abort caused by: failed to send RPC: sending to all 1 replicas failed; last error: (err: node unavailable; try another peer) <nil>
W200519 14:07:10.955277 485542 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: EOF:
I200519 14:07:10.957735 457927 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200519 14:07:10.961096 1188672 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:40013: context canceled
I200519 14:07:10.961605 459786 rpc/nodedialer/nodedialer.go:160  [n5,liveness-hb,txn=439245d1] unable to connect to n2: failed to connect to n2 at 127.0.0.1:44743: context canceled
I200519 14:07:10.968535 459844 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (split): node unavailable; try another peer
W200519 14:07:10.968975 455747 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
I200519 14:07:10.970753 485544 kv/kvserver/replica_destroy.go:217  [n4,s4,r33/14:{m-/Table/Syste…}] removed 5 (0+5) keys in 349ms [clear=347ms commit=2ms]
I200519 14:07:10.973811 1188755 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:37739: context canceled
E200519 14:07:10.974250 453648 kv/kvserver/reports/reporter.go:139  [n1,replication-reporter] failed to generate replication reports: failed to save locality report: failed to send RPC: sending to all 1 replicas failed; last error: (err: node unavailable; try another peer) <nil>
I200519 14:07:10.975747 1188743 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:44743: context canceled
W200519 14:07:10.976388 456688 ts/db.go:194  [n3,ts-poll] error writing time series data: failed to send RPC: sending to all 3 replicas failed; last error: (err: node unavailable; try another peer) <nil>
I200519 14:07:10.977972 459786 rpc/nodedialer/nodedialer.go:160  [n5,liveness-hb,txn=439245d1] unable to connect to n4: failed to connect to n4 at 127.0.0.1:42969: context canceled
I200519 14:07:10.978618 459844 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicate): node unavailable; try another peer
E200519 14:07:10.984627 1188242 kv/kvserver/queue.go:1079  [n4,raftlog,s4,r3/3:/System/{NodeLive…-tsd}] result is ambiguous (server shutdown)
W200519 14:07:10.985496 458229 server/node.go:718  [n4,summaries] error recording status summaries: result is ambiguous (server shutdown)
W200519 14:07:10.985689 485544 kv/kvserver/store.go:1841  [n4,s4,r33/14:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
I200519 14:07:10.988372 459844 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200519 14:07:10.989845 459844 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200519 14:07:10.990221 459844 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200519 14:07:10.990516 459844 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
W200519 14:07:10.989283 459786 kv/txn.go:607  [n5,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (error=failed to connect to n4 at 127.0.0.1:42969: context canceled [exhausted])
I200519 14:07:10.990848 459844 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200519 14:07:10.991264 459786 kv/kvserver/node_liveness.go:832  [n5,liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (error=failed to connect to n4 at 127.0.0.1:42969: context canceled [exhausted])
W200519 14:07:10.996319 459786 kv/txn.go:607  [n5,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200519 14:07:10.997136 459786 kv/kvserver/node_liveness.go:496  [n5,liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200519 14:07:11.039350 458005 ts/db.go:194  [n4,ts-poll] error writing time series data: node unavailable; try another peer
I200519 14:07:11.122383 1188774 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:37739: context canceled
W200519 14:07:11.202335 457900 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200519 14:07:11.255119 457900 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200519 14:07:11.343629 457900 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200519 14:07:11.496623 459785 ts/db.go:197  [n5,ts-poll] node unavailable; try another peer
E200519 14:07:11.634224 1188797 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestReplicateQueueDownReplicate (457.55s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):5, (n3,s3):4, (n4,s4):14LEARNER, next=15, gen=32, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):5, (n3,s3):4, next=15, gen=33, sticky=9223372036.854775807,2147483647]
        goroutine 453658 [running]:
        runtime/debug.Stack(0xc060507a00, 0x72db540, 0xc0328f3200)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7401200, 0xc044e6ff00, 0xc060507a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc044e6ff00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x5ef
        testing.tRunner(0xc044e6ff00, 0x6489db8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1955899&tab=buildLog) on [master@8fec3f4c6d136a86f472c975edd36b75e5ab9a8c](https://github.com/cockroachdb/cockroach/commits/8fec3f4c6d136a86f472c975edd36b75e5ab9a8c):

```
E200520 05:53:50.208492 1252672 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.209114 1252673 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200520 05:53:50.209173 443895 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200520 05:53:50.225565 1252674 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.225698 1252591 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.226942 1252635 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.411766 1252641 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.413033 1252642 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.427215 1252757 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200520 05:53:50.508763 441247 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200520 05:53:50.538051 1252758 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.538850 1252728 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.539435 1252717 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.564012 1252698 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.565606 1252718 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200520 05:53:50.579069 444971 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200520 05:53:50.579597 441089 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
W200520 05:53:50.601504 443895 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200520 05:53:50.605787 1252761 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.606416 1252690 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.606465 1252729 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.627982 1252803 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200520 05:53:50.636334 444971 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200520 05:53:50.673512 441288 server/status/runtime.go:499  [n1] runtime stats: 12 GiB RSS, 657 goroutines, 1.4 GiB/143 MiB/1.7 GiB GO alloc/idle/total(stale), 282 MiB/387 MiB CGO alloc/total, 3480.4 CGO/sec, 342.6/31.3 %(u/s)time, 0.0 %gc (0x), 1.8 MiB/1.8 MiB (r/w)net
E200520 05:53:50.682926 1252784 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200520 05:53:50.692032 443002 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
W200520 05:53:50.721898 441289 ts/db.go:197  [n1,ts-poll] node unavailable; try another peer
E200520 05:53:50.739798 1252806 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.741876 1252785 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200520 05:53:50.743908 444971 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200520 05:53:50.766011 1252807 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.807960 1252736 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.831514 1252832 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:50.883772 1252765 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W200520 05:53:50.955141 444971 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200520 05:53:50.964554 1252814 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:51.006078 1252833 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:51.007092 1252751 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:51.012680 1252846 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:51.025410 1252834 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I200520 05:53:51.030869 441278 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 160 iterations on up to 9145 requests
E200520 05:53:51.162296 1252931 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:51.163181 1252869 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E200520 05:53:51.163699 1252847 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestReplicateQueueDownReplicate (521.39s)
    testing.go:853: race detected during execution of test


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1961474&tab=buildLog) on [master@d7985c53f1e6d32abd604f5987b9a44f6b210ee3](https://github.com/cockroachdb/cockroach/commits/d7985c53f1e6d32abd604f5987b9a44f6b210ee3):

```
W200522 06:55:42.722161 454764 kv/txn.go:607  [n5,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
I200522 06:55:42.723086 1198048 kv/kvserver/node_liveness.go:832  [n3,s3,r5/5:{/Systemtse-m}] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (error=failed to connect to n4 at 127.0.0.1:38707: context canceled [exhausted])
E200522 06:55:42.725755 1198600 kv/kvserver/replica_range_lease.go:383  [n5,s5,r5/4:{/Systemtse-m}] mismatch incrementing epoch for {NodeID:4 Epoch:1 Expiration:1590130539.494450574,0 Draining:false Decommissioning:false}; actual is {NodeID:4 Epoch:1 Expiration:1590130543.955704127,0 Draining:false Decommissioning:false}
E200522 06:55:42.726326 1198972 kv/kvserver/replica_range_lease.go:339  [n1,s1,r6/1:/Table/{SystemCon…-11}] node unavailable; try another peer
W200522 06:55:42.731479 451252 kv/kvserver/store.go:1652  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200522 06:55:42.739318 454745 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
E200522 06:55:42.739957 1198778 kv/kvserver/replica_range_lease.go:383  [n2,s2,r18/5:/Table/2{2-3}] node unavailable; try another peer
E200522 06:55:42.749909 1198577 kv/kvserver/replica_range_lease.go:339  [n5,s5,r16/2:/Table/2{0-1}] node unavailable; try another peer
W200522 06:55:42.750529 454764 kv/kvserver/node_liveness.go:571  [n5,liveness-hb] slow heartbeat took 1.8s
E200522 06:55:42.765963 1198048 kv/kvserver/replica_range_lease.go:383  [n3,s3,r5/5:{/Systemtse-m}] mismatch incrementing epoch for {NodeID:4 Epoch:1 Expiration:1590130539.494450574,0 Draining:false Decommissioning:false}; actual is {NodeID:4 Epoch:1 Expiration:1590130543.955704127,0 Draining:false Decommissioning:false}
I200522 06:55:42.767827 454745 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
W200522 06:55:42.773887 449008 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:42.779579 454764 kv/kvserver/node_liveness.go:496  [n5,liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
I200522 06:55:42.783350 454745 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
E200522 06:55:42.790241 449214 jobs/registry.go:507  error while adopting jobs: failed querying for jobs: adopt-job: failed to send RPC: sending to all 1 replicas failed; last error: (err: node unavailable; try another peer) <nil>
W200522 06:55:42.842463 454658 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:42.844169 451252 kv/kvserver/store.go:1652  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:42.889283 454658 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:42.900228 449008 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:42.946914 451252 kv/kvserver/store.go:1652  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:42.954808 452977 ts/db.go:194  [n4,ts-poll] error writing time series data: result is ambiguous (server shutdown)
W200522 06:55:42.996045 454658 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.124995 449008 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.155004 451252 kv/kvserver/store.go:1652  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.178076 451844 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.224386 451844 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.230085 454658 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.342412 451844 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.538858 451844 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.554490 451252 kv/kvserver/store.go:1652  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.557810 449008 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200522 06:55:43.599494 454658 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200522 06:55:43.742268 449054 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 115 iterations on up to 8597 requests
--- FAIL: TestReplicateQueueDownReplicate (458.50s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n5,s5):13LEARNER, next=14, gen=30, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n4,s4):3, next=14, gen=31, sticky=9223372036.854775807,2147483647]
        goroutine 448856 [running]:
        runtime/debug.Stack(0xc018401a28, 0x7870f40, 0xc048826760)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x799bba0, 0xc01187ab00, 0xc018401a28)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc01187ab00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc01187ab00, 0x695a830)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1962885&tab=buildLog) on [master@d5dd67bc846b73cebe8dc73b320de045ce8a170c](https://github.com/cockroachdb/cockroach/commits/d5dd67bc846b73cebe8dc73b320de045ce8a170c):

```
W200523 06:31:36.273971 447124 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: EOF:
W200523 06:31:36.274588 447028 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200523 06:31:36.277827 510601 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: rpc error: code = Unavailable desc = transport is closing:
W200523 06:31:36.285070 510255 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200523 06:31:36.287133 547633 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200523 06:31:36.287550 1092533 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42885: context canceled
W200523 06:31:36.300445 443544 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
I200523 06:31:36.304019 1092494 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42305: context canceled
W200523 06:31:36.306492 445710 kv/kvserver/store.go:1652  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200523 06:31:36.307291 1092507 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42305: context canceled
I200523 06:31:36.307797 1092447 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39109: context canceled
W200523 06:31:36.310210 510296 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200523 06:31:36.310678 1092508 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39109: context canceled
I200523 06:31:36.310936 1092495 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42189: context canceled
I200523 06:31:36.311150 446331 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200523 06:31:36.313032 1092497 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42885: context canceled
W200523 06:31:36.314319 464403 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200523 06:31:36.315025 529265 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 2: EOF:
I200523 06:31:36.316179 1092531 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39109: context canceled
I200523 06:31:36.316498 1092509 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42885: context canceled
I200523 06:31:36.318518 1092446 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39153: context canceled
I200523 06:31:36.307822 1092496 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39153: context canceled
I200523 06:31:36.320959 1092448 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42189: context canceled
I200523 06:31:36.321066 1092498 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39153: context canceled
W200523 06:31:36.319483 547556 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Unavailable desc = transport is closing:
W200523 06:31:36.322600 446547 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200523 06:31:36.324145 445447 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
I200523 06:31:36.329729 446331 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (gc): node unavailable; try another peer
W200523 06:31:36.331718 528517 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200523 06:31:36.332368 510717 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200523 06:31:36.332998 548452 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200523 06:31:36.350234 449305 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200523 06:31:36.351746 1092532 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42189: context canceled
I200523 06:31:36.356128 1092552 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39109: context canceled
I200523 06:31:36.356337 1092550 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42305: context canceled
I200523 06:31:36.359963 1092549 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42885: context canceled
W200523 06:31:36.361191 445710 kv/kvserver/store.go:1652  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200523 06:31:36.366929 1092551 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39153: context canceled
I200523 06:31:36.368248 449237 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200523 06:31:36.368969 449237 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200523 06:31:36.377277 447476 ts/db.go:194  [n4,ts-poll] error writing time series data: failed to send RPC: sending to all 3 replicas failed; last error: (err: node unavailable; try another peer) <nil>
W200523 06:31:36.402321 449187 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200523 06:31:36.449436 449187 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200523 06:31:36.485386 1092629 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42189: context canceled
--- FAIL: TestReplicateQueueDownReplicate (416.45s)
    testing.go:853: race detected during execution of test


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1964894&tab=buildLog) on [master@809829bfe7ff27a610fa78f409fae658b9f2d9d9](https://github.com/cockroachdb/cockroach/commits/809829bfe7ff27a610fa78f409fae658b9f2d9d9):

```
W200524 06:06:55.709523 449346 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: rpc error: code = Unavailable desc = transport is closing:
I200524 06:06:55.712523 1038437 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43587: context canceled
W200524 06:06:55.699795 443267 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 06:06:55.700517 442916 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 06:06:55.719188 441412 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200524 06:06:55.720214 1038337 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43587: context canceled
I200524 06:06:55.722493 444023 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (gc): node unavailable; try another peer
W200524 06:06:55.722770 450027 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: EOF:
I200524 06:06:55.723130 444023 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (merge): node unavailable; try another peer
W200524 06:06:55.723385 444309 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200524 06:06:55.723672 444023 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (split): node unavailable; try another peer
I200524 06:06:55.724234 444023 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200524 06:06:55.724478 444023 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200524 06:06:55.724823 444023 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
W200524 06:06:55.727692 449693 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: EOF:
W200524 06:06:55.728475 445740 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 06:06:55.704201 442950 gossip/gossip.go:1525  [n5] no incoming or outgoing connections
W200524 06:06:55.735489 511654 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 2: EOF:
I200524 06:06:55.735627 1038375 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:35807: context canceled
I200524 06:06:55.736058 1038436 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42603: context canceled
W200524 06:06:55.736512 535969 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 06:06:55.739231 442639 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 06:06:55.741744 443973 kv/kvserver/store.go:1648  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200524 06:06:55.741937 1038372 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39225: context canceled
W200524 06:06:55.743704 445056 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 06:06:55.746384 440633 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200524 06:06:55.747679 590008 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 06:06:55.698646 524390 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: rpc error: code = Unavailable desc = transport is closing:
I200524 06:06:55.752093 1038374 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42603: context canceled
I200524 06:06:55.753473 1038335 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:35807: context canceled
W200524 06:06:55.753674 510976 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200524 06:06:55.758879 1038338 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39225: context canceled
I200524 06:06:55.759240 1038336 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42603: context canceled
I200524 06:06:55.762267 1038373 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39753: context canceled
W200524 06:06:55.764067 511099 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200524 06:06:55.769471 1038390 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:35807: context canceled
I200524 06:06:55.781585 1038391 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43587: context canceled
I200524 06:06:55.782686 1038273 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39753: context canceled
I200524 06:06:55.783316 1038272 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39225: context canceled
W200524 06:06:55.785631 440290 kv/kvserver/store.go:1648  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200524 06:06:55.790425 443973 kv/kvserver/store.go:1648  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200524 06:06:55.852256 1038395 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43587: context canceled
I200524 06:06:55.854250 1038397 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42603: context canceled
W200524 06:06:55.907554 443973 kv/kvserver/store.go:1648  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (348.74s)
    testing.go:853: race detected during execution of test


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1966144&tab=buildLog) on [master@533095b4f6d74439d6c11c0b57f5c7b80937fb5c](https://github.com/cockroachdb/cockroach/commits/533095b4f6d74439d6c11c0b57f5c7b80937fb5c):

```
I200524 18:07:27.256960 1077327 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:34163: context canceled
I200524 18:07:27.261727 1077625 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:44217: context canceled
I200524 18:07:27.267900 1077622 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:46629: context canceled
I200524 18:07:27.270846 1077623 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:44971: context canceled
W200524 18:07:27.274700 540028 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: EOF:
W200524 18:07:27.284814 523535 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: EOF:
I200524 18:07:27.285548 1077572 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:44217: context canceled
W200524 18:07:27.288562 457711 gossip/gossip.go:1525  [n5] no incoming or outgoing connections
W200524 18:07:27.294582 456506 kv/kvserver/store.go:1648  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200524 18:07:27.295764 523534 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: EOF:
I200524 18:07:27.306134 1077624 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:34163: context canceled
W200524 18:07:27.310794 523966 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: EOF:
I200524 18:07:27.330128 1077537 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:46629: context canceled
I200524 18:07:27.332611 1077538 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33621: context canceled
I200524 18:07:27.339051 1077652 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:44217: context canceled
W200524 18:07:27.343407 538689 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 18:07:27.404658 456277 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 18:07:27.408959 523622 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200524 18:07:27.410396 455848 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
I200524 18:07:27.411005 1077500 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:34163: context canceled
W200524 18:07:27.432470 456506 kv/kvserver/store.go:1648  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200524 18:07:27.436003 1077502 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:46629: context canceled
I200524 18:07:27.440316 1077503 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:44971: context canceled
I200524 18:07:27.560512 1077722 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33621: context canceled
I200524 18:07:27.563669 1077723 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:46629: context canceled
I200524 18:07:27.581404 1077724 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:34163: context canceled
I200524 18:07:27.597924 1077640 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33621: context canceled
W200524 18:07:27.613139 456506 kv/kvserver/store.go:1648  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200524 18:07:27.631639 455968 gossip/gossip.go:1525  [n4] no incoming or outgoing connections
W200524 18:07:27.696804 455498 ts/db.go:194  [n3,ts-poll] error writing time series data: node unavailable; try another peer
I200524 18:07:27.821138 452699 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 113 iterations on up to 8071 requests
I200524 18:07:27.824895 456573 server/status/runtime.go:499  [n4] runtime stats: 12 GiB RSS, 659 goroutines, 1.4 GiB/271 MiB/1.7 GiB GO alloc/idle/total(stale), 346 MiB/456 MiB CGO alloc/total, 3705.0 CGO/sec, 322.4/27.3 %(u/s)time, 0.3 %gc (1x), 1.9 MiB/1.9 MiB (r/w)net
W200524 18:07:27.937595 456574 ts/db.go:197  [n4,ts-poll] node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (385.47s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n5,s5):4, (n2,s2):2, (n3,s3):5, (n4,s4):11LEARNER, next=12, gen=26, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n5,s5):4, (n2,s2):2, (n3,s3):5, next=12, gen=27, sticky=9223372036.854775807,2147483647]
        goroutine 452533 [running]:
        runtime/debug.Stack(0xc05371da28, 0x786e8a0, 0xc024e7ba60)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x799c440, 0xc015950900, 0xc05371da28)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc015950900)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc015950900, 0x6971c28)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1967936&tab=buildLog) on [master@8f7cd615bc0cf910816d81905c0e683f9748c77d](https://github.com/cockroachdb/cockroach/commits/8f7cd615bc0cf910816d81905c0e683f9748c77d):

```
W200526 06:34:23.676462 453799 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200526 06:34:23.876851 832209 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: EOF:
I200526 06:34:23.883725 1181128 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45487: context canceled
I200526 06:34:23.665915 448650 rpc/nodedialer/nodedialer.go:160  [n1,ts-poll] unable to connect to n2: failed to connect to n2 at 127.0.0.1:38499: context canceled
W200526 06:34:23.905178 452593 kv/kvserver/store.go:1648  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200526 06:34:23.907569 1181132 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:41785: context canceled
W200526 06:34:23.908227 448650 ts/db.go:194  [n1,ts-poll] error writing time series data: failed to send RPC: sending to all 3 replicas failed; last error: <nil> failed to connect to n5 at 127.0.0.1:46093: context canceled
W200526 06:34:23.667940 683327 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200526 06:34:23.909193 450725 server/node.go:718  [n2,summaries] error recording status summaries: node unavailable; try another peer
W200526 06:34:23.889350 515984 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: EOF:
W200526 06:34:23.890523 585450 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: EOF:
W200526 06:34:23.893387 452200 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200526 06:34:23.896634 1181133 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:38499: context canceled
W200526 06:34:23.914598 517363 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200526 06:34:23.917759 517298 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200526 06:34:23.920163 1181200 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:46093: context canceled
I200526 06:34:23.926441 1181202 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45487: context canceled
I200526 06:34:23.929179 1181219 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:38499: context canceled
W200526 06:34:23.963406 451570 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200526 06:34:23.964209 455762 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Unavailable desc = transport is closing:
W200526 06:34:23.964936 452593 kv/kvserver/store.go:1648  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200526 06:34:24.025686 1181286 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45487: context canceled
I200526 06:34:24.039749 1181287 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:37953: context canceled
I200526 06:34:24.053756 1181051 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:37953: context canceled
W200526 06:34:24.058020 452593 kv/kvserver/store.go:1648  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 06:34:24.234918 450842 ts/db.go:197  [n2,ts-poll] node unavailable; try another peer
W200526 06:34:24.258972 452593 kv/kvserver/store.go:1648  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200526 06:34:24.350357 448466 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 125 iterations on up to 8745 requests
I200526 06:34:24.385090 450841 server/status/runtime.go:499  [n2] runtime stats: 12 GiB RSS, 575 goroutines, 1.5 GiB/82 MiB/1.7 GiB GO alloc/idle/total(stale), 296 MiB/392 MiB CGO alloc/total, 4009.2 CGO/sec, 339.2/34.2 %(u/s)time, 0.0 %gc (0x), 1.4 MiB/1.4 MiB (r/w)net
W200526 06:34:24.525887 451617 server/node.go:718  [n3,summaries] error recording status summaries: node unavailable; try another peer
W200526 06:34:24.618978 452593 kv/kvserver/store.go:1648  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 06:34:24.687099 454331 kv/kvserver/store.go:1648  [n5,s5,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 06:34:24.734190 454331 kv/kvserver/store.go:1648  [n5,s5,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (443.53s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n5,s5):4, (n2,s2):16LEARNER, next=17, gen=36, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n5,s5):4, next=17, gen=37, sticky=9223372036.854775807,2147483647]
        goroutine 448429 [running]:
        runtime/debug.Stack(0xc044a1fa28, 0x6f7c700, 0xc0518fbe80)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x70aa2c0, 0xc0102be200, 0xc044a1fa28)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc0102be200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc0102be200, 0x607f838)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1969213&tab=buildLog) on [master@0857bf6d080b857a31b2a60846d8ce406289b584](https://github.com/cockroachdb/cockroach/commits/0857bf6d080b857a31b2a60846d8ce406289b584):

```
W200526 12:24:11.857354 448209 kv/kvserver/store.go:1648  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:11.858907 450368 kv/kvserver/store.go:1648  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:11.885249 451905 kv/kvserver/store_raft.go:509  [n4,s4,r4/2:/System{/tsd-tse}] handle raft ready: 2.7s [applied=9, batches=3, state_assertions=0]
W200526 12:24:11.897913 451892 kv/kvserver/store_raft.go:509  [n4,s4,r32/5:/{Table/36-Max}] handle raft ready: 1.8s [applied=1, batches=1, state_assertions=1]
I200526 12:24:11.913520 454819 kv/kvserver/replica_destroy.go:217  [n4,s4,r33/7:{m-/Table/Syste…}] removed 5 (0+5) keys in 904ms [clear=891ms commit=13ms]
W200526 12:24:11.956722 454819 kv/kvserver/store.go:1837  [n4,s4,r33/7:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
W200526 12:24:11.978256 454783 kv/kvserver/store.go:1648  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.015767 448134 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
W200526 12:24:12.052756 450368 kv/kvserver/store.go:1648  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.057236 451911 kv/kvserver/store_raft.go:509  [n4,s4,r3/2:/System/{NodeLive…-tsd}] handle raft ready: 3.0s [applied=13, batches=8, state_assertions=3]
W200526 12:24:12.074931 448209 kv/kvserver/store.go:1648  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.161801 450913 kv/kvserver/store.go:1648  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.211353 451896 kv/kvserver/store_raft.go:509  [n4,s4,r2/3:/System/NodeLiveness{-Max}] handle raft ready: 3.2s [applied=38, batches=12, state_assertions=4]
W200526 12:24:12.217704 450913 kv/kvserver/store.go:1648  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.296497 451971 kv/kvserver/store.go:1648  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.369280 451971 kv/kvserver/store.go:1648  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.375006 450913 kv/kvserver/store.go:1648  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.405398 452012 kv/kvserver/node_liveness.go:571  [n4,liveness-hb] slow heartbeat took 3.2s
W200526 12:24:12.431786 454783 kv/kvserver/store.go:1648  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.445611 1436822 kv/kvserver/node_liveness.go:571  [n4,s4,r26/3:/NamespaceTable/{30-Max}] slow heartbeat took 2.7s
W200526 12:24:12.458816 450368 kv/kvserver/store.go:1648  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.461160 448209 kv/kvserver/store.go:1648  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.484308 451971 kv/kvserver/store.go:1648  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.570824 450913 kv/kvserver/store.go:1648  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.667795 451971 kv/kvserver/store.go:1648  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:12.917973 448283 server/node.go:718  [n1,summaries] error recording status summaries: node unavailable; try another peer
W200526 12:24:12.962860 450913 kv/kvserver/store.go:1648  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:13.019283 451971 kv/kvserver/store.go:1648  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:13.237059 454783 kv/kvserver/store.go:1648  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:13.268432 450368 kv/kvserver/store.go:1648  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:13.362564 448209 kv/kvserver/store.go:1648  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 12:24:13.494570 448141 ts/db.go:197  [n1,ts-poll] node unavailable; try another peer
I200526 12:24:13.573884 448218 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 123 iterations on up to 10409 requests
--- FAIL: TestReplicateQueueDownReplicate (691.54s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n2,s2):5, (n4,s4):7LEARNER, next=8, gen=18, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n2,s2):5, next=8, gen=19, sticky=9223372036.854775807,2147483647]
        goroutine 447970 [running]:
        runtime/debug.Stack(0xc01ae4da28, 0x786eb80, 0xc02b03b740)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x799c740, 0xc020b49e00, 0xc01ae4da28)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc020b49e00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc020b49e00, 0x6971cb8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1969255&tab=buildLog) on [master@8f7cd615bc0cf910816d81905c0e683f9748c77d](https://github.com/cockroachdb/cockroach/commits/8f7cd615bc0cf910816d81905c0e683f9748c77d):

```
I200526 13:28:22.765085 1244855 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39247: context canceled
W200526 13:28:22.766820 521215 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200526 13:28:22.770870 462591 kv/txn.go:607  [n2,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
I200526 13:28:22.771207 1244370 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:36145: context canceled
I200526 13:28:22.780005 1244545 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39247: context canceled
I200526 13:28:22.786115 1244546 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:37111: context canceled
I200526 13:28:22.788209 1244916 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:44825: context canceled
W200526 13:28:22.800040 462591 kv/kvserver/node_liveness.go:496  [n2,liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
I200526 13:28:22.812118 1244872 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:36145: context canceled
I200526 13:28:22.826549 460507 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200526 13:28:22.827623 460507 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200526 13:28:22.857739 464950 kv/kvserver/store_remove_replica.go:122  [n4,s4,r33/13:{m-/Table/Syste…}] removing replica r33/13
W200526 13:28:22.859528 464950 kv/kvserver/store.go:1837  [n4,s4,r33/13:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
W200526 13:28:22.860105 464950 kv/kvserver/store_raft.go:509  [n4,s4,r33/13:{m-/Table/Syste…}] handle raft ready: 0.6s [applied=1, batches=1, state_assertions=0]
I200526 13:28:22.864580 463546 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (gc): node unavailable; try another peer
E200526 13:28:22.868847 1244183 kv/kvserver/queue.go:1079  [n3,raftlog,s3,r3/2:/System/{NodeLive…-tsd}] result is ambiguous (server shutdown)
W200526 13:28:22.875455 466618 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: node unavailable; try another peer:
W200526 13:28:22.993828 460408 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
W200526 13:28:23.119773 462858 ts/db.go:194  [n2,ts-poll] error writing time series data: node unavailable; try another peer
W200526 13:28:23.137730 463520 kv/kvserver/store.go:1648  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 13:28:23.203431 463520 kv/kvserver/store.go:1648  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 13:28:23.286812 466865 kv/kvserver/store.go:1648  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 13:28:23.305671 463520 kv/kvserver/store.go:1648  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 13:28:23.352990 466865 kv/kvserver/store.go:1648  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200526 13:28:23.454606 465116 server/status/runtime.go:499  [n4] runtime stats: 12 GiB RSS, 657 goroutines, 1.6 GiB/134 MiB/1.8 GiB GO alloc/idle/total(stale), 296 MiB/400 MiB CGO alloc/total, 4674.5 CGO/sec, 349.5/30.0 %(u/s)time, 0.0 %gc (0x), 1.6 MiB/1.6 MiB (r/w)net
W200526 13:28:23.468848 466865 kv/kvserver/store.go:1648  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 13:28:23.489464 463520 kv/kvserver/store.go:1648  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 13:28:23.674441 466865 kv/kvserver/store.go:1648  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 13:28:23.768144 465117 ts/db.go:194  [n4,ts-poll] error writing time series data: result is ambiguous (server shutdown)
W200526 13:28:23.768670 465117 ts/db.go:197  [n4,ts-poll] node unavailable; try another peer
E200526 13:28:23.777925 467568 jobs/registry.go:507  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
W200526 13:28:23.910382 467516 server/node.go:718  [n5,summaries] error recording status summaries: node unavailable; try another peer
W200526 13:28:24.119951 466865 kv/kvserver/store.go:1648  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (526.69s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n5,s5):5, (n4,s4):13LEARNER, next=14, gen=30, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n5,s5):5, next=14, gen=31, sticky=9223372036.854775807,2147483647]
        goroutine 460316 [running]:
        runtime/debug.Stack(0xc02b9d7a28, 0x786eb80, 0xc02cf2e320)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x799c740, 0xc047310b00, 0xc02b9d7a28)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc047310b00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc047310b00, 0x6971cb8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1969789&tab=buildLog) on [master@9620bb9e538a83ad54f8b12f06eec32bc094325c](https://github.com/cockroachdb/cockroach/commits/9620bb9e538a83ad54f8b12f06eec32bc094325c):

```
I200526 18:35:39.370352 1383915 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:43509: context canceled
I200526 18:35:39.373862 1383913 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:33227: context canceled
I200526 18:35:39.381488 462221 kv/kvserver/replica_destroy.go:217  [n5,s5,r33/12:{m-/Table/Syste…}] removed 5 (0+5) keys in 321ms [clear=308ms commit=13ms]
W200526 18:35:39.382686 462221 kv/kvserver/store.go:1837  [n5,s5,r33/12:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
W200526 18:35:39.391050 462219 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 1: EOF:
W200526 18:35:39.395665 455591 ts/db.go:194  [n2,ts-poll] error writing time series data: node unavailable; try another peer
I200526 18:35:39.397457 459469 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200526 18:35:39.405235 459469 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200526 18:35:39.406813 459469 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (split): node unavailable; try another peer
I200526 18:35:39.407693 459469 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200526 18:35:39.408666 459469 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200526 18:35:39.409547 459469 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200526 18:35:39.413341 459469 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200526 18:35:39.414424 1384035 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:41191: context canceled
I200526 18:35:39.414929 1384009 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:40275: context canceled
I200526 18:35:39.417976 1384011 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:33227: context canceled
I200526 18:35:39.410536 1384010 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:41191: context canceled
I200526 18:35:39.397849 1384008 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39963: context canceled
W200526 18:35:39.403550 590268 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Unavailable desc = transport is closing:
I200526 18:35:39.421420 1384036 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39963: context canceled
W200526 18:35:39.403292 514213 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: EOF:
I200526 18:35:39.425582 459469 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200526 18:35:39.426108 459469 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200526 18:35:39.426516 457624 server/node.go:718  [n4,summaries] error recording status summaries: node unavailable; try another peer
I200526 18:35:39.411454 1383906 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:40275: context canceled
W200526 18:35:39.445926 457379 kv/kvserver/store.go:1648  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 18:35:39.446201 660203 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 4: EOF:
W200526 18:35:39.453931 513123 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200526 18:35:39.458562 659968 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200526 18:35:39.474686 1383794 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:41191: context canceled
I200526 18:35:39.475459 1383791 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:33227: context canceled
I200526 18:35:39.479404 1383792 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:43509: context canceled
W200526 18:35:39.498093 457379 kv/kvserver/store.go:1648  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (605.58s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):7, (n3,s3):3, (n5,s5):12LEARNER, next=13, gen=31, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):7, (n3,s3):3, next=13, gen=32, sticky=9223372036.854775807,2147483647]
        goroutine 453138 [running]:
        runtime/debug.Stack(0xc0505d7a28, 0x788fb40, 0xc0144da380)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x79bdd80, 0xc01360d600, 0xc0505d7a28)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc01360d600)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc01360d600, 0x69797b8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1970208&tab=buildLog) on [master@6f92e589a933ded4b7d3c34eb38918aa29a999a8](https://github.com/cockroachdb/cockroach/commits/6f92e589a933ded4b7d3c34eb38918aa29a999a8):

```
W200526 21:03:33.324902 459900 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200526 21:03:33.325873 504341 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: rpc error: code = Unavailable desc = transport is closing:
W200526 21:03:33.333335 452514 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200526 21:03:33.333894 1242093 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:35293: context canceled
I200526 21:03:33.337664 1242094 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:45267: context canceled
W200526 21:03:33.340106 455091 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: rpc error: code = Unavailable desc = transport is closing:
I200526 21:03:33.345059 1242096 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:43875: context canceled
I200526 21:03:33.347100 1241951 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:38139: context canceled
W200526 21:03:33.356762 491766 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200526 21:03:33.360125 450727 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200526 21:03:33.365635 450342 kv/kvserver/store.go:1652  [n2,s2,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200526 21:03:33.393819 1242108 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:38139: context canceled
W200526 21:03:33.400071 447985 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
I200526 21:03:33.513685 1242011 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:38139: context canceled
W200526 21:03:33.546681 450342 kv/kvserver/store.go:1652  [n2,s2,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 21:03:33.548691 452514 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 21:03:33.613811 449993 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
W200526 21:03:33.737222 448112 kv/kvserver/store.go:1652  [n1,s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200526 21:03:33.738591 448111 kv/kvserver/store.go:1652  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
I200526 21:03:33.746736 448120 gossip/gossip.go:577  [n1] gossip status (ok, 5 nodes)
gossip client (0/3 cur/max conns)
gossip server (0/3 cur/max conns, infos 7600/2590 sent/received, bytes 1502279B/525185B sent/received)
W200526 21:03:33.792227 448111 kv/kvserver/store.go:1652  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200526 21:03:33.795913 448112 kv/kvserver/store.go:1652  [n1,s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200526 21:03:33.890436 448111 kv/kvserver/store.go:1652  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200526 21:03:33.895204 448112 kv/kvserver/store.go:1652  [n1,s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200526 21:03:33.906147 451013 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 21:03:33.953091 451013 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 21:03:33.958742 448235 server/node.go:718  [n1,summaries] error recording status summaries: node unavailable; try another peer
W200526 21:03:33.973605 450342 kv/kvserver/store.go:1652  [n2,s2,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 21:03:33.975616 452514 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200526 21:03:34.041739 451013 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200526 21:03:34.043905 448036 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 173 iterations on up to 9080 requests
--- FAIL: TestReplicateQueueDownReplicate (480.71s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n5,s5):3, (n3,s3):15LEARNER, next=16, gen=34, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):2, (n5,s5):3, next=16, gen=35, sticky=9223372036.854775807,2147483647]
        goroutine 447890 [running]:
        runtime/debug.Stack(0xc010323a28, 0x788c760, 0xc0141531c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x79baa00, 0xc041286c00, 0xc010323a28)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc041286c00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc041286c00, 0x69798e0)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1972349&tab=buildLog) on [master@d04627c975965916f6794503cb2c51530949ef7c](https://github.com/cockroachdb/cockroach/commits/d04627c975965916f6794503cb2c51530949ef7c):

```
W200527 14:55:09.148649 454923 ts/db.go:194  [n4,ts-poll] error writing time series data: node unavailable; try another peer
W200527 14:55:09.163086 457011 kv/kvserver/store.go:1652  [n5,s5,r6/4:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200527 14:55:09.175195 452862 kv/kvserver/store.go:1652  [n2,s2,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 14:55:09.180334 457012 kv/kvserver/store.go:1652  [n5,s5,r2/2:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200527 14:55:09.248132 457359 server/node.go:718  [n5,summaries] error recording status summaries: node unavailable; try another peer
W200527 14:55:09.254658 457011 kv/kvserver/store.go:1652  [n5,s5,r6/4:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
I200527 14:55:09.255636 452880 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200527 14:55:09.256404 452880 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200527 14:55:09.293253 457012 kv/kvserver/store.go:1652  [n5,s5,r2/2:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200527 14:55:09.331643 450366 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
W200527 14:55:09.434650 453557 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 14:55:09.464727 457012 kv/kvserver/store.go:1652  [n5,s5,r2/2:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200527 14:55:09.479855 453557 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 14:55:09.485904 457011 kv/kvserver/store.go:1652  [n5,s5,r6/4:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200527 14:55:09.547619 452862 kv/kvserver/store.go:1652  [n2,s2,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 14:55:09.576864 453557 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 14:55:09.647916 452550 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
I200527 14:55:09.674772 450459 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200527 14:55:09.675687 450459 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200527 14:55:09.763354 453557 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200527 14:55:09.775151 453031 jobs/registry.go:507  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
W200527 14:55:09.843375 456786 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 14:55:09.862619 457012 kv/kvserver/store.go:1652  [n5,s5,r2/2:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200527 14:55:09.872755 457011 kv/kvserver/store.go:1652  [n5,s5,r6/4:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200527 14:55:09.892639 456786 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200527 14:55:09.916581 457024 server/status/runtime.go:499  [n5] runtime stats: 12 GiB RSS, 654 goroutines, 1.4 GiB/264 MiB/1.7 GiB GO alloc/idle/total(stale), 366 MiB/471 MiB CGO alloc/total, 3362.4 CGO/sec, 361.8/34.7 %(u/s)time, 0.0 %gc (0x), 996 KiB/996 KiB (r/w)net
W200527 14:55:10.007130 456786 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 14:55:10.166485 453557 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 14:55:10.224733 456786 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 14:55:10.335576 452862 kv/kvserver/store.go:1652  [n2,s2,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200527 14:55:10.458134 450450 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 111 iterations on up to 8073 requests
W200527 14:55:10.497624 457025 ts/db.go:197  [n5,ts-poll] node unavailable; try another peer
W200527 14:55:10.584667 454908 kv/kvserver/store.go:1652  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (427.78s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n3,s3):5, (n4,s4):13LEARNER, next=14, gen=30, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n3,s3):5, next=14, gen=31, sticky=9223372036.854775807,2147483647]
        goroutine 450324 [running]:
        runtime/debug.Stack(0xc01734da28, 0x7861ce0, 0xc0335897a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7990660, 0xc011a8e800, 0xc01734da28)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc011a8e800)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc011a8e800, 0x694a6e8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1973236&tab=buildLog) on [master@e6e96a48d5f941f1d71c3c7d79665b89f4950615](https://github.com/cockroachdb/cockroach/commits/e6e96a48d5f941f1d71c3c7d79665b89f4950615):

```
I200527 22:03:33.855660 460938 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (split): node unavailable; try another peer
I200527 22:03:33.857292 460938 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200527 22:03:33.857713 460938 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200527 22:03:33.857901 460938 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200527 22:03:33.859232 460938 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200527 22:03:33.859408 460938 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200527 22:03:33.859811 460938 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200527 22:03:33.889937 456681 kv/kvserver/store.go:1652  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200527 22:03:33.911309 1129689 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39395: context canceled
I200527 22:03:33.913344 1129690 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:45451: context canceled
I200527 22:03:33.913853 1129691 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42865: context canceled
I200527 22:03:33.920138 1129692 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39007: context canceled
I200527 22:03:33.943548 1129629 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:38315: context canceled
I200527 22:03:33.944375 1129630 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:45451: context canceled
I200527 22:03:33.944505 1129628 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39007: context canceled
I200527 22:03:33.980043 1129743 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39395: context canceled
I200527 22:03:33.981004 1129746 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42865: context canceled
W200527 22:03:33.982180 457804 gossip/gossip.go:1525  [n4] no incoming or outgoing connections
W200527 22:03:33.989037 456681 kv/kvserver/store.go:1652  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:33.989361 454433 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.034034 454433 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.143386 454433 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.199923 456681 kv/kvserver/store.go:1652  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.206601 454425 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
W200527 22:03:34.327429 454433 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.387596 457301 kv/kvserver/store.go:1652  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.438720 457301 kv/kvserver/store.go:1652  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.488666 460886 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.528328 457301 kv/kvserver/store.go:1652  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.537263 460886 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.619332 456681 kv/kvserver/store.go:1652  [n2,s2,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200527 22:03:34.650666 460886 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200527 22:03:34.662861 454604 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 112 iterations on up to 8579 requests
--- FAIL: TestReplicateQueueDownReplicate (419.58s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):4, (n5,s5):5, (n4,s4):12LEARNER, next=13, gen=28, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):4, (n5,s5):5, next=13, gen=29, sticky=9223372036.854775807,2147483647]
        goroutine 454303 [running]:
        runtime/debug.Stack(0xc045701a18, 0x7864c40, 0xc028075740)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7993640, 0xc047280c00, 0xc045701a18)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc047280c00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc047280c00, 0x694c7d8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1973652&tab=buildLog) on [master@28496b58d6e04dc061c272b872a659876068db18](https://github.com/cockroachdb/cockroach/commits/28496b58d6e04dc061c272b872a659876068db18):

```
W200528 02:52:06.265472 458350 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200528 02:52:06.266676 457640 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200528 02:52:06.287885 457640 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200528 02:52:06.268965 459161 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200528 02:52:06.269239 546843 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200528 02:52:06.269418 455507 rpc/nodedialer/nodedialer.go:160  [n1,liveness-hb,txn=4f07414f] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42621: context canceled
I200528 02:52:06.292954 455507 rpc/nodedialer/nodedialer.go:160  [n1,liveness-hb,txn=4f07414f] unable to connect to n4: failed to connect to n4 at 127.0.0.1:33869: context canceled
I200528 02:52:06.297744 455507 rpc/nodedialer/nodedialer.go:160  [n1,liveness-hb,txn=4f07414f] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33173: context canceled
W200528 02:52:06.299016 455507 kv/txn.go:607  [n1,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (error=failed to connect to n2 at 127.0.0.1:33173: context canceled [exhausted])
I200528 02:52:06.299366 455507 kv/kvserver/node_liveness.go:832  [n1,liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (error=failed to connect to n2 at 127.0.0.1:33173: context canceled [exhausted])
W200528 02:52:06.299893 461483 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 02:52:06.301751 455507 kv/txn.go:607  [n1,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
I200528 02:52:06.302200 457640 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (gc): node unavailable; try another peer
W200528 02:52:06.302600 455507 kv/kvserver/node_liveness.go:496  [n1,liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200528 02:52:06.269682 546845 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200528 02:52:06.269977 457331 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
I200528 02:52:06.308482 1109848 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:33869: context canceled
I200528 02:52:06.302903 457640 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200528 02:52:06.311374 457640 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (split): node unavailable; try another peer
I200528 02:52:06.311967 457640 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200528 02:52:06.315367 457640 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200528 02:52:06.316064 457640 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200528 02:52:06.316530 457640 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200528 02:52:06.323124 1109861 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:38727: context canceled
I200528 02:52:06.323558 1109863 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33173: context canceled
W200528 02:52:06.325599 457422 ts/db.go:194  [n2,ts-poll] error writing time series data: failed to send RPC: sending to all 3 replicas failed; last error: <nil> failed to connect to n5 at 127.0.0.1:44211: context canceled
W200528 02:52:06.331873 458324 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 02:52:06.355874 458358 ts/db.go:194  [n3,ts-poll] error writing time series data: result is ambiguous (server shutdown)
W200528 02:52:06.402132 461892 server/node.go:718  [n5,summaries] error recording status summaries: node unavailable; try another peer
W200528 02:52:06.412197 461483 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200528 02:52:06.425966 461597 server/status/runtime.go:499  [n5] runtime stats: 11 GiB RSS, 559 goroutines, 1.5 GiB/76 MiB/1.7 GiB GO alloc/idle/total(stale), 137 MiB/195 MiB CGO alloc/total, 4744.5 CGO/sec, 354.4/30.9 %(u/s)time, 0.0 %gc (1x), 1.8 MiB/1.8 MiB (r/w)net
W200528 02:52:06.601251 461598 ts/db.go:197  [n5,ts-poll] node unavailable; try another peer
W200528 02:52:06.601999 461483 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (406.59s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: snapshot failed: (n4,s4):17LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: raft group deleted
        goroutine 455305 [running]:
        runtime/debug.Stack(0xc00de09a18, 0x786efe0, 0xc027ebda00)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x799dbc0, 0xc02faa1400, 0xc00de09a18)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc02faa1400)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc02faa1400, 0x6954d08)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1973850&tab=buildLog) on [master@fed9db2eb997cc2dea177bf5c37629b37c919c92](https://github.com/cockroachdb/cockroach/commits/fed9db2eb997cc2dea177bf5c37629b37c919c92):

```
W200528 04:51:12.574395 533167 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200528 04:51:12.584993 1153822 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42185: context canceled
W200528 04:51:12.586444 454129 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 4: EOF:
I200528 04:51:12.588051 1153823 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33445: context canceled
W200528 04:51:12.589575 524609 kv/kvserver/store.go:1841  [n3,s3,r33/13:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
W200528 04:51:12.593808 514146 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: EOF:
W200528 04:51:12.600053 693669 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: EOF:
W200528 04:51:12.600436 453595 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: EOF:
I200528 04:51:12.603577 1153360 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33445: context canceled
I200528 04:51:12.606659 1153361 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:44261: context canceled
W200528 04:51:12.607576 448198 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 04:51:12.608512 456053 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200528 04:51:12.610426 454070 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 04:51:12.622566 524607 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200528 04:51:12.630658 1153362 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45053: context canceled
I200528 04:51:12.631496 1153923 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42185: context canceled
I200528 04:51:12.634448 1153824 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39643: context canceled
I200528 04:51:12.634990 1153825 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:44261: context canceled
W200528 04:51:12.640633 450918 kv/kvserver/store.go:1652  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 04:51:12.647863 451572 gossip/gossip.go:1525  [n4] no incoming or outgoing connections
I200528 04:51:12.660435 1153899 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42185: context canceled
W200528 04:51:12.662796 453860 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200528 04:51:12.675227 1153834 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45053: context canceled
I200528 04:51:12.678249 1153832 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33445: context canceled
W200528 04:51:12.681519 454070 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200528 04:51:12.681815 1153831 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42185: context canceled
I200528 04:51:12.691600 1153833 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39643: context canceled
W200528 04:51:12.693886 450918 kv/kvserver/store.go:1652  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 04:51:12.714160 448198 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 04:51:12.812805 454070 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 04:51:12.814175 450918 kv/kvserver/store.go:1652  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 04:51:13.033921 454070 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 04:51:13.380336 454070 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (410.76s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n2,s2):3, (n3,s3):13LEARNER, next=14, gen=30, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n2,s2):3, next=14, gen=31, sticky=9223372036.854775807,2147483647]
        goroutine 448083 [running]:
        runtime/debug.Stack(0xc04593da18, 0x7876f60, 0xc02e6833c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x79a5ea0, 0xc024f01100, 0xc04593da18)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc024f01100)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc024f01100, 0x695c1d8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1974992&tab=buildLog) on [master@96ec98f19ea5a971fe38b68cf29baed6cde8cb26](https://github.com/cockroachdb/cockroach/commits/96ec98f19ea5a971fe38b68cf29baed6cde8cb26):

```
W200528 11:31:40.835313 452437 kv/txn.go:607  [n1,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200528 11:31:40.836012 452437 kv/kvserver/node_liveness.go:496  [n1,liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200528 11:31:40.840741 1263550 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200528 11:31:40.845834 1208624 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200528 11:31:40.845890 461407 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 2: EOF:
I200528 11:31:40.879367 1293060 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45855: context canceled
I200528 11:31:40.846560 1292943 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45855: context canceled
E200528 11:31:40.881093 1292115 kv/kvserver/queue.go:1079  [n4,raftlog,s4,r4/3:/System{/tsd-tse}] result is ambiguous (server shutdown)
W200528 11:31:40.847280 460207 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200528 11:31:40.847551 461408 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200528 11:31:40.884331 459981 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200528 11:31:40.884821 455907 gossip/gossip.go:1525  [n4] no incoming or outgoing connections
W200528 11:31:40.886210 455374 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200528 11:31:40.897241 1292994 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:40253: context canceled
I200528 11:31:40.899662 1293004 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:40253: context canceled
I200528 11:31:40.901694 1293005 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39285: context canceled
I200528 11:31:40.905254 1293006 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:33989: context canceled
I200528 11:31:40.907743 1293061 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39285: context canceled
I200528 11:31:40.908423 1293007 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45105: context canceled
I200528 11:31:40.918424 1293108 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45855: context canceled
I200528 11:31:40.925240 1293059 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:33989: context canceled
I200528 11:31:40.930045 1293133 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45855: context canceled
I200528 11:31:40.931324 1293136 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39285: context canceled
W200528 11:31:40.964350 457627 ts/db.go:194  [n5,ts-poll] error writing time series data: node unavailable; try another peer
W200528 11:31:40.987647 456256 ts/db.go:194  [n4,ts-poll] error writing time series data: result is ambiguous (server shutdown)
W200528 11:31:41.021676 455374 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 11:31:41.238463 455374 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 11:31:41.538179 452348 gossip/gossip.go:1525  [n1] no incoming or outgoing connections
W200528 11:31:41.609390 455374 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 11:31:41.677393 455969 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 11:31:41.743464 455969 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200528 11:31:41.817904 452225 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 165 iterations on up to 8728 requests
W200528 11:31:41.839931 455969 kv/kvserver/store.go:1652  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (479.25s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: snapshot failed: (n4,s4):15LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: raft group deleted
        goroutine 452178 [running]:
        runtime/debug.Stack(0xc03490ba18, 0x7876ca0, 0xc0136293a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x79a5be0, 0xc0329f2500, 0xc03490ba18)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc0329f2500)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x678
        testing.tRunner(0xc0329f2500, 0x695be58)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1975092&tab=buildLog) on [master@e16ca57dac107d389ea317ec0239f45f5a2cc139](https://github.com/cockroachdb/cockroach/commits/e16ca57dac107d389ea317ec0239f45f5a2cc139):

```
I200528 13:16:50.067959 466502 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200528 13:16:50.073806 464417 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (split): node unavailable; try another peer
E200528 13:16:50.075072 1101111 kv/kvclient/kvcoord/txn_interceptor_committer.go:449  making txn commit explicit failed for "unnamed" meta={id=5751850e key=/Table/28/1/1/0 pri=0.04104637 epo=0 ts=1590671806.753830430,0 min=1590671806.753830430,0 seq=2} lock=true stat=STAGING rts=1590671806.753830430,0 wto=false max=1590671807.253830430,0 ifw=1: result is ambiguous (error=failed to connect to n1 at 127.0.0.1:34893: context canceled [exhausted])
I200528 13:16:50.078267 463265 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200528 13:16:50.079817 464417 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200528 13:16:50.078598 466502 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200528 13:16:50.080797 466502 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200528 13:16:50.080986 464417 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200528 13:16:50.079881 463265 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200528 13:16:50.082464 464417 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200528 13:16:50.083139 463265 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200528 13:16:50.084213 463265 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200528 13:16:50.084766 463265 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200528 13:16:50.083584 464417 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200528 13:16:50.085523 463265 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200528 13:16:50.086088 464417 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200528 13:16:50.086467 460437 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200528 13:16:50.087406 464417 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200528 13:16:50.087565 460437 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200528 13:16:50.091721 466502 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200528 13:16:50.092225 466502 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200528 13:16:50.092690 466502 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (split): node unavailable; try another peer
I200528 13:16:50.092944 466502 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200528 13:16:50.093227 466502 kv/kvserver/queue.go:579  [n5,s5] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200528 13:16:50.107385 460437 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200528 13:16:50.107966 460437 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200528 13:16:50.108511 460437 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (split): node unavailable; try another peer
W200528 13:16:50.108997 463326 ts/db.go:197  [n3,ts-poll] node unavailable; try another peer
I200528 13:16:50.109640 460437 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200528 13:16:50.110349 460437 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200528 13:16:50.112730 460437 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
W200528 13:16:50.126825 460426 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200528 13:16:50.140090 462725 ts/db.go:194  [n2,ts-poll] error writing time series data: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (394.60s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):5, (n5,s5):3, (n4,s4):13LEARNER, next=14, gen=30, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n2,s2):5, (n5,s5):3, next=14, gen=31, sticky=9223372036.854775807,2147483647]
        goroutine 460221 [running]:
        runtime/debug.Stack(0xc0150439a0, 0x787cb20, 0xc021de33a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x79abae0, 0xc02e85fe00, 0xc0150439a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc02e85fe00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc02e85fe00, 0x695c9b8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1976327&tab=buildLog) on [master@ca57e95a2b97eff03ee13d5977685eff4a1dd905](https://github.com/cockroachdb/cockroach/commits/ca57e95a2b97eff03ee13d5977685eff4a1dd905):

```
I200529 00:50:00.393567 457522 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (split): node unavailable; try another peer
I200529 00:50:00.473223 457522 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200529 00:50:00.474044 457522 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200529 00:50:00.474828 457522 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200529 00:50:00.475467 457522 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200529 00:50:00.476011 457522 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200529 00:50:00.476802 457522 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200529 00:50:00.394067 463612 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200529 00:50:00.471525 1410070 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:33657: context canceled
W200529 00:50:00.397390 514749 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200529 00:50:00.480979 1409976 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:36255: context canceled
I200529 00:50:00.400615 1410031 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:36255: context canceled
I200529 00:50:00.488459 1409974 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39779: context canceled
I200529 00:50:00.489063 1409977 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:35995: context canceled
W200529 00:50:00.415141 456761 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200529 00:50:00.417326 456386 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200529 00:50:00.422754 1409883 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39779: context canceled
W200529 00:50:00.430181 528159 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200529 00:50:00.435207 514866 kv/kvserver/replica_destroy.go:164  [n5,s5,r33/12:{m-/Table/Syste…}] removed 6 (0+6) keys in 555ms [clear=507ms commit=48ms]
W200529 00:50:00.492908 514866 kv/kvserver/store.go:1841  [n5,s5,r33/12:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
W200529 00:50:00.493970 514864 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200529 00:50:00.436336 1410032 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39779: context canceled
I200529 00:50:00.440923 1410052 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:33657: context canceled
W200529 00:50:00.392954 459051 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
I200529 00:50:00.494820 1409979 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:35995: context canceled
W200529 00:50:00.501509 455758 server/node.go:718  [n2,summaries] error recording status summaries: node unavailable; try another peer
W200529 00:50:00.404713 460324 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: EOF:
W200529 00:50:00.535860 456386 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200529 00:50:00.576056 459464 kv/kvserver/store.go:1652  [n5,s5,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200529 00:50:00.641966 456386 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200529 00:50:00.678233 455563 server/status/runtime.go:499  [n2] runtime stats: 12 GiB RSS, 655 goroutines, 1.5 GiB/176 MiB/1.8 GiB GO alloc/idle/total(stale), 212 MiB/278 MiB CGO alloc/total, 3476.9 CGO/sec, 321.2/24.8 %(u/s)time, 0.0 %gc (0x), 2.3 MiB/2.3 MiB (r/w)net
W200529 00:50:00.691813 455564 ts/db.go:194  [n2,ts-poll] error writing time series data: node unavailable; try another peer
W200529 00:50:00.776683 459464 kv/kvserver/store.go:1652  [n5,s5,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (634.29s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n2,s2):5, (n3,s3):2, (n4,s4):3, (n5,s5):12LEARNER, next=13, gen=28, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n2,s2):5, (n3,s3):2, (n4,s4):3, next=13, gen=29, sticky=9223372036.854775807,2147483647]
        goroutine 453368 [running]:
        runtime/debug.Stack(0xc014d5b9a0, 0x78a5740, 0xc02915d960)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x79d4ce0, 0xc0136a0000, 0xc014d5b9a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc0136a0000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc0136a0000, 0x69795f0)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1979907&tab=buildLog) on [master@021d87644837dfc7e7cd56af386f94b93f41623e](https://github.com/cockroachdb/cockroach/commits/021d87644837dfc7e7cd56af386f94b93f41623e):

```
W200530 11:56:41.875735 456588 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200530 11:56:41.876443 451635 rpc/nodedialer/nodedialer.go:160  [n4,ts-poll] unable to connect to n1: failed to connect to n1 at 127.0.0.1:45195: context canceled
W200530 11:56:41.877686 451635 ts/db.go:194  [n4,ts-poll] error writing time series data: failed to send RPC: sending to all 3 replicas failed; last error: <nil> failed to connect to n1 at 127.0.0.1:45195: context canceled
I200530 11:56:41.887671 1203004 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:36449: context canceled
W200530 11:56:41.888171 1121993 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200530 11:56:41.888926 506094 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200530 11:56:41.893819 450468 ts/db.go:194  [n3,ts-poll] error writing time series data: result is ambiguous (server shutdown)
I200530 11:56:41.902540 1203012 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42361: context canceled
I200530 11:56:41.904988 1203003 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38419: context canceled
W200530 11:56:41.907716 453095 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200530 11:56:41.907985 1203005 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36797: context canceled
W200530 11:56:41.919545 450910 gossip/gossip.go:1525  [n4] no incoming or outgoing connections
W200530 11:56:41.928148 451252 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200530 11:56:41.929842 447358 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200530 11:56:41.934031 454056 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200530 11:56:41.936913 1203010 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38419: context canceled
W200530 11:56:41.938142 450758 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200530 11:56:41.940374 531730 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200530 11:56:41.942006 1203006 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36797: context canceled
I200530 11:56:41.956689 1203007 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42361: context canceled
I200530 11:56:41.966656 1203060 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42361: context canceled
I200530 11:56:41.971718 1203061 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36797: context canceled
W200530 11:56:41.992375 447358 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200530 11:56:41.994064 454056 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200530 11:56:42.053169 1203050 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:45195: context canceled
W200530 11:56:42.090549 454056 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200530 11:56:42.109255 447358 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200530 11:56:42.193497 451556 kv/kvserver/store.go:1652  [n4,s4,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200530 11:56:42.207752 447233 server/node.go:718  [n1,summaries] error recording status summaries: node unavailable; try another peer
W200530 11:56:42.254154 451556 kv/kvserver/store.go:1652  [n4,s4,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200530 11:56:42.283206 447358 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200530 11:56:42.303252 454056 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200530 11:56:42.369881 451556 kv/kvserver/store.go:1652  [n4,s4,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (451.02s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):4, (n5,s5):3, (n3,s3):13LEARNER, next=14, gen=30, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):4, (n5,s5):3, next=14, gen=31, sticky=9223372036.854775807,2147483647]
        goroutine 447149 [running]:
        runtime/debug.Stack(0xc0415119a0, 0x7a476a0, 0xc00fc9e9a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7b7d0e0, 0xc04ba40f00, 0xc0415119a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc04ba40f00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc04ba40f00, 0x6af59d8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1980235&tab=buildLog) on [master@7bf340170868117c910eb327414def1f5583bb70](https://github.com/cockroachdb/cockroach/commits/7bf340170868117c910eb327414def1f5583bb70):

```
W200531 02:01:27.127486 456177 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200531 02:01:27.129234 1161637 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:38543: context canceled
I200531 02:01:27.131260 449813 rpc/nodedialer/nodedialer.go:160  [n1,ts-poll] unable to connect to n2: failed to connect to n2 at 127.0.0.1:34587: context canceled
W200531 02:01:27.133906 449813 ts/db.go:194  [n1,ts-poll] error writing time series data: failed to send RPC: sending to all 3 replicas failed; last error: <nil> failed to connect to n2 at 127.0.0.1:34587: context canceled
I200531 02:01:27.137498 1161636 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:40481: context canceled
W200531 02:01:27.144986 453453 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200531 02:01:27.151966 1161468 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36605: context canceled
I200531 02:01:27.155661 519253 kv/kvserver/replica_destroy.go:164  [n5,s5,r33/12:{m-/Table/Syste…}] removed 5 (0+5) keys in 395ms [clear=328ms commit=67ms]
W200531 02:01:27.156728 519253 kv/kvserver/store.go:1841  [n5,s5,r33/12:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
I200531 02:01:27.159123 1161423 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:34587: context canceled
I200531 02:01:27.173906 1161424 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:38543: context canceled
W200531 02:01:27.175118 518658 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: EOF:
W200531 02:01:27.176164 520003 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: node unavailable; try another peer:
W200531 02:01:27.176264 452702 kv/kvserver/store.go:1652  [n3,s3,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200531 02:01:27.052603 610389 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 5: EOF:
I200531 02:01:27.200745 1161617 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:41287: context canceled
I200531 02:01:27.073752 453935 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200531 02:01:27.208638 454939 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200531 02:01:27.212943 453935 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200531 02:01:27.216424 453935 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200531 02:01:27.218577 453935 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (split): node unavailable; try another peer
I200531 02:01:27.220258 453935 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200531 02:01:27.221003 453935 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
W200531 02:01:27.214577 610388 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200531 02:01:27.222678 456177 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200531 02:01:27.230585 453935 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200531 02:01:27.231858 453935 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200531 02:01:27.261642 1161716 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36605: context canceled
W200531 02:01:27.290959 452043 ts/db.go:194  [n2,ts-poll] error writing time series data: node unavailable; try another peer
I200531 02:01:27.338968 1161703 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:41287: context canceled
W200531 02:01:27.407556 452702 kv/kvserver/store.go:1652  [n3,s3,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200531 02:01:27.430307 456177 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200531 02:01:27.665590 449818 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 121 iterations on up to 8250 requests
--- FAIL: TestReplicateQueueDownReplicate (409.28s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n3,s3):4, (n5,s5):12LEARNER, next=13, gen=28, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n3,s3):4, next=13, gen=29, sticky=9223372036.854775807,2147483647]
        goroutine 449595 [running]:
        runtime/debug.Stack(0xc03ce179a0, 0x7a466a0, 0xc044091020)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7b7c0e0, 0xc0176a2b00, 0xc03ce179a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc0176a2b00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc0176a2b00, 0x6af4e28)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1981890&tab=buildLog) on [master@1422dd50c19eaab5e1766c56b9502c7838f17756](https://github.com/cockroachdb/cockroach/commits/1422dd50c19eaab5e1766c56b9502c7838f17756):

```
I200531 11:54:42.557281 1154600 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39117: context canceled
I200531 11:54:42.560349 1154601 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:45541: context canceled
I200531 11:54:42.563513 1154603 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38431: context canceled
W200531 11:54:42.563764 452123 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200531 11:54:42.567139 1154582 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:45541: context canceled
I200531 11:54:42.567570 1154602 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:45299: context canceled
W200531 11:54:42.570926 535094 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200531 11:54:42.572104 1154604 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:41447: context canceled
W200531 11:54:42.573327 493453 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200531 11:54:42.575709 452975 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
I200531 11:54:42.576930 1154556 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38431: context canceled
I200531 11:54:42.577609 449467 server/status/runtime.go:499  [n1] runtime stats: 11 GiB RSS, 740 goroutines, 1.3 GiB/239 MiB/1.6 GiB GO alloc/idle/total(stale), 162 MiB/221 MiB CGO alloc/total, 3126.7 CGO/sec, 321.7/27.9 %(u/s)time, 0.1 %gc (1x), 2.1 MiB/2.1 MiB (r/w)net
W200531 11:54:42.578254 452401 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200531 11:54:42.579270 453484 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200531 11:54:42.580050 456598 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200531 11:54:42.582289 490738 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200531 11:54:42.583150 453125 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200531 11:54:42.586890 1154606 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:45541: context canceled
I200531 11:54:42.588489 1154557 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:41447: context canceled
I200531 11:54:42.592363 1154558 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39117: context canceled
I200531 11:54:42.604776 1154559 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:45541: context canceled
W200531 11:54:42.651350 449468 ts/db.go:197  [n1,ts-poll] node unavailable; try another peer
W200531 11:54:42.666584 449623 server/node.go:718  [n1,summaries] error recording status summaries: node unavailable; try another peer
W200531 11:54:42.667985 452123 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200531 11:54:42.694201 1154651 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38431: context canceled
I200531 11:54:42.704457 1154652 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:41447: context canceled
I200531 11:54:42.755544 1154498 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39117: context canceled
I200531 11:54:42.758269 1154708 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:45299: context canceled
W200531 11:54:42.863771 452123 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200531 11:54:43.255172 452123 kv/kvserver/store.go:1652  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200531 11:54:43.388733 449248 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 153 iterations on up to 8359 requests
W200531 11:54:43.393394 453458 kv/kvserver/store.go:1652  [n4,s4,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200531 11:54:43.456275 453458 kv/kvserver/store.go:1652  [n4,s4,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (431.71s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):4, (n3,s3):5, (n4,s4):11LEARNER, next=12, gen=26, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):4, (n3,s3):5, next=12, gen=27, sticky=9223372036.854775807,2147483647]
        goroutine 449320 [running]:
        runtime/debug.Stack(0xc04767b9a0, 0x7a46820, 0xc02c18d7e0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7b7c260, 0xc00e473e00, 0xc04767b9a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc00e473e00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc00e473e00, 0x6af4f68)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1984714&tab=buildLog) on [master@d65b224bc7c8b99effd42f14677aa066ddc94cdb](https://github.com/cockroachdb/cockroach/commits/d65b224bc7c8b99effd42f14677aa066ddc94cdb):

```
I200602 06:44:46.076419 1245312 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:37353: context canceled
I200602 06:44:46.079037 1245314 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:35893: context canceled
I200602 06:44:46.089499 1245248 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:37353: context canceled
W200602 06:44:46.097925 457121 ts/db.go:194  [n4,ts-poll] error writing time series data: failed to send RPC: sending to all 3 replicas failed; last error: <nil> failed to connect to n3 at 127.0.0.1:34387: context canceled
W200602 06:44:46.106071 454917 ts/db.go:197  [n2,ts-poll] node unavailable; try another peer
I200602 06:44:46.110390 457156 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (split): node unavailable; try another peer
W200602 06:44:46.113729 506956 kv/kvserver/store.go:1841  [n5,s5,r33/10:{m-/Table/Syste…}] unable to gossip on capacity change: node unavailable; try another peer
W200602 06:44:46.114950 457121 ts/db.go:197  [n4,ts-poll] node unavailable; try another peer
I200602 06:44:46.115926 457156 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200602 06:44:46.118758 457156 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200602 06:44:46.119124 457156 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200602 06:44:46.119474 457156 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200602 06:44:46.119685 457156 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200602 06:44:46.119947 457156 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200602 06:44:46.160389 458935 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:46.264586 458935 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:46.312538 455671 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:46.426964 457253 server/node.go:718  [n4,summaries] error recording status summaries: node unavailable; try another peer
W200602 06:44:46.445135 458935 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:46.512747 452586 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:46.571220 452586 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:46.677082 452586 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:46.731589 455671 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:46.788606 458935 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:46.902153 452586 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200602 06:44:46.913604 457265 sql/stats/automatic_stats.go:300  failed to refresh stats: node unavailable; try another peer
W200602 06:44:47.036379 457058 kv/kvserver/store.go:1652  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:47.073175 454870 kv/kvserver/store.go:1652  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 06:44:47.088158 457058 kv/kvserver/store.go:1652  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200602 06:44:47.096486 452621 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 117 iterations on up to 8933 requests
W200602 06:44:47.140931 454870 kv/kvserver/store.go:1652  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200602 06:44:47.170005 457120 server/status/runtime.go:499  [n4] runtime stats: 12 GiB RSS, 397 goroutines, 1.4 GiB/209 MiB/1.7 GiB GO alloc/idle/total(stale), 87 MiB/137 MiB CGO alloc/total, 3741.8 CGO/sec, 325.5/28.5 %(u/s)time, 0.1 %gc (1x), 1.8 MiB/1.8 MiB (r/w)net
W200602 06:44:47.203717 457058 kv/kvserver/store.go:1652  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (516.43s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n3,s3):4, (n5,s5):10LEARNER, next=11, gen=24, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n3,s3):4, next=11, gen=25, sticky=9223372036.854775807,2147483647]
        goroutine 452422 [running]:
        runtime/debug.Stack(0xc04811f9a0, 0x7a51680, 0xc03cd6e6a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7b87de0, 0xc0379e8100, 0xc04811f9a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc0379e8100)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc0379e8100, 0x6ad29f0)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1984789&tab=buildLog) on [master@e80ccc59e22823a4e5a353d08d66c9ba16c1525d](https://github.com/cockroachdb/cockroach/commits/e80ccc59e22823a4e5a353d08d66c9ba16c1525d):

```
I200602 07:03:14.428643 461859 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200602 07:03:14.428870 461859 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200602 07:03:14.432047 468465 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200602 07:03:14.439651 464972 gossip/gossip.go:1525  [n4] no incoming or outgoing connections
I200602 07:03:14.445144 1175580 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:35175: context canceled
I200602 07:03:14.446400 461859 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200602 07:03:14.456373 1175590 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33971: context canceled
W200602 07:03:14.462313 461841 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 07:03:14.469644 540853 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200602 07:03:14.474761 1175457 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43327: context canceled
W200602 07:03:14.490553 467512 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E200602 07:03:14.494799 464610 jobs/registry.go:501  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
W200602 07:03:14.494938 468743 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200602 07:03:14.498743 467137 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200602 07:03:14.500182 468679 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200602 07:03:14.500786 463804 ts/db.go:197  [n2,ts-poll] node unavailable; try another peer
W200602 07:03:14.504390 467088 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200602 07:03:14.515949 1175645 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:42663: context canceled
W200602 07:03:14.518188 461841 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 07:03:14.522851 465397 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200602 07:03:14.525120 464734 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200602 07:03:14.528039 461891 ts/db.go:194  [n1,ts-poll] error writing time series data: result is ambiguous (server shutdown)
E200602 07:03:14.543065 464118 sql/stats/automatic_stats.go:300  failed to refresh stats: node unavailable; try another peer
W200602 07:03:14.547633 467512 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200602 07:03:14.615601 1175553 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:41463: context canceled
I200602 07:03:14.619634 1175554 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43327: context canceled
I200602 07:03:14.625568 1175684 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:42663: context canceled
W200602 07:03:14.634774 464454 kv/kvserver/store.go:1652  [n3,s3,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 07:03:14.650004 467512 kv/kvserver/store.go:1652  [n5,s5,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200602 07:03:14.663054 1175671 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33971: context canceled
I200602 07:03:14.664034 1175672 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:41463: context canceled
I200602 07:03:14.665931 1175670 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:35175: context canceled
E200602 07:03:14.736342 466067 jobs/registry.go:501  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (452.96s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n4,s4):3, (n3,s3):9LEARNER, next=10, gen=22, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n5,s5):2, (n4,s4):3, next=10, gen=23, sticky=9223372036.854775807,2147483647]
        goroutine 461599 [running]:
        runtime/debug.Stack(0xc0288c99a0, 0x7a4c7c0, 0xc0389c27e0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7b82e20, 0xc0118fdd00, 0xc0288c99a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc0118fdd00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc0118fdd00, 0x6acf9b8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1984999&tab=buildLog) on [master@cbe7fdc0d1ebfdef71967172d5a8843ca3f86229](https://github.com/cockroachdb/cockroach/commits/cbe7fdc0d1ebfdef71967172d5a8843ca3f86229):

```
I200602 13:35:49.354809 453298 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200602 13:35:49.355230 453298 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200602 13:35:49.355785 453298 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200602 13:35:49.348074 463685 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 5: EOF:
W200602 13:35:49.363361 453113 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
I200602 13:35:49.365703 453402 rpc/nodedialer/nodedialer.go:160  [n2,liveness-hb,txn=907c32b3] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43207: context canceled
I200602 13:35:49.368884 453402 rpc/nodedialer/nodedialer.go:160  [n2,liveness-hb,txn=907c32b3] unable to connect to n4: failed to connect to n4 at 127.0.0.1:34955: context canceled
W200602 13:35:49.373091 457525 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200602 13:35:49.379428 453402 rpc/nodedialer/nodedialer.go:160  [n2,liveness-hb,txn=907c32b3] unable to connect to n3: failed to connect to n3 at 127.0.0.1:40243: context canceled
W200602 13:35:49.380603 453402 kv/txn.go:607  [n2,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (error=failed to connect to n3 at 127.0.0.1:40243: context canceled [exhausted])
I200602 13:35:49.381081 453402 kv/kvserver/node_liveness.go:832  [n2,liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (error=failed to connect to n3 at 127.0.0.1:40243: context canceled [exhausted])
W200602 13:35:49.383533 453402 kv/txn.go:607  [n2,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200602 13:35:49.384254 453402 kv/kvserver/node_liveness.go:496  [n2,liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200602 13:35:49.398064 774823 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200602 13:35:49.400833 454863 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200602 13:35:49.421981 513933 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200602 13:35:49.425404 476665 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: rpc error: code = Unavailable desc = transport is closing:
W200602 13:35:49.428384 454632 gossip/gossip.go:1525  [n4] no incoming or outgoing connections
W200602 13:35:49.445495 514067 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
E200602 13:35:49.446741 1183037 kv/kvserver/queue.go:1079  [n3,replicaGC,s3,r33/11:{m-/Table/Syste…}] aborted in distSender: context canceled
W200602 13:35:49.447383 466797 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200602 13:35:49.469700 1183935 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43207: context canceled
I200602 13:35:49.472418 1183937 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:40243: context canceled
I200602 13:35:49.473753 1183936 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:34955: context canceled
I200602 13:35:49.477587 1183938 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39121: context canceled
I200602 13:35:49.557972 1184172 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39121: context canceled
I200602 13:35:49.559090 1184171 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:40243: context canceled
I200602 13:35:49.561872 1184173 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43207: context canceled
I200602 13:35:49.564881 1184174 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:40409: context canceled
E200602 13:35:49.580070 456002 jobs/registry.go:501  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
W200602 13:35:49.810603 457621 kv/kvserver/store.go:1652  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 13:35:49.857486 457621 kv/kvserver/store.go:1652  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 13:35:49.972976 457621 kv/kvserver/store.go:1652  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (462.56s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n2,s2):4, (n3,s3):11LEARNER, next=12, gen=26, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):2, (n2,s2):4, next=12, gen=27, sticky=9223372036.854775807,2147483647]
        goroutine 450975 [running]:
        runtime/debug.Stack(0xc03f33d9a0, 0x7a59160, 0xc0564d89e0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7b8fb20, 0xc03afed700, 0xc03f33d9a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc03afed700)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc03afed700, 0x6ad57d0)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1985493&tab=buildLog) on [master@b4b7d454a3ccd3e5d9c2e25010aaf61a681993e0](https://github.com/cockroachdb/cockroach/commits/b4b7d454a3ccd3e5d9c2e25010aaf61a681993e0):

```
I200602 18:28:28.032124 1030553 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:35273: context canceled
I200602 18:28:28.034097 1030503 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:39153: context canceled
W200602 18:28:28.032832 464737 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200602 18:28:28.033093 1030401 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:39153: context canceled
W200602 18:28:28.033286 460292 kv/kvserver/node_liveness.go:496  [n2,liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
I200602 18:28:28.033486 1030504 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42139: context canceled
W200602 18:28:28.033685 464713 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 2: EOF:
W200602 18:28:28.033890 512461 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 3: EOF:
I200602 18:28:28.035834 1030505 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:33969: context canceled
I200602 18:28:28.036408 460274 server/status/runtime.go:499  [n2] runtime stats: 11 GiB RSS, 1182 goroutines, 1.3 GiB/307 MiB/1.7 GiB GO alloc/idle/total(stale), 154 MiB/213 MiB CGO alloc/total, 3093.7 CGO/sec, 308.5/25.8 %(u/s)time, 0.0 %gc (1x), 2.0 MiB/2.0 MiB (r/w)net
I200602 18:28:28.036632 1030431 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:33969: context canceled
I200602 18:28:28.036929 1030311 rpc/nodedialer/nodedialer.go:160  [n4,replicaGC,s4,r33/12:{m-/Table/Syste…}] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36469: context canceled
W200602 18:28:28.037112 466643 kv/kvserver/raft_transport.go:637  [n5] while processing outgoing Raft queue to node 4: EOF:
I200602 18:28:28.037306 1030430 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42139: context canceled
W200602 18:28:28.037908 468933 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: EOF:
I200602 18:28:28.041379 1030311 rpc/nodedialer/nodedialer.go:160  [n4,replicaGC,s4,r33/12:{m-/Table/Syste…}] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42139: context canceled
I200602 18:28:28.043962 1030556 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42139: context canceled
I200602 18:28:28.048804 1030311 rpc/nodedialer/nodedialer.go:160  [n4,replicaGC,s4,r33/12:{m-/Table/Syste…}] unable to connect to n1: failed to connect to n1 at 127.0.0.1:39153: context canceled
W200602 18:28:28.050125 463709 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 1: EOF:
E200602 18:28:28.050564 1030311 kv/kvserver/queue.go:1079  [n4,replicaGC,s4,r33/12:{m-/Table/Syste…}] failed to send RPC: sending to all 5 replicas failed; last error: <nil> failed to connect to n1 at 127.0.0.1:39153: context canceled
I200602 18:28:28.052308 1030554 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36469: context canceled
I200602 18:28:28.067943 1030514 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:39153: context canceled
W200602 18:28:28.081252 459979 gossip/gossip.go:1525  [n2] no incoming or outgoing connections
W200602 18:28:28.086055 461318 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200602 18:28:28.151544 1030648 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:35273: context canceled
W200602 18:28:28.156950 463152 gossip/gossip.go:1525  [n5] no incoming or outgoing connections
E200602 18:28:28.208016 458157 jobs/registry.go:501  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
I200602 18:28:28.253394 1030653 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42139: context canceled
I200602 18:28:28.257864 1030654 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:33969: context canceled
I200602 18:28:28.260452 1030652 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36469: context canceled
W200602 18:28:28.495201 464234 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 18:28:28.553323 464234 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200602 18:28:28.640035 464234 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (363.75s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):5, (n2,s2):3, (n4,s4):12LEARNER, next=13, gen=28, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):5, (n2,s2):3, next=13, gen=29, sticky=9223372036.854775807,2147483647]
        goroutine 457800 [running]:
        runtime/debug.Stack(0xc00b46b9a0, 0x7a5ad20, 0xc04430d5e0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7b916a0, 0xc012a03c00, 0xc00b46b9a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc012a03c00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc012a03c00, 0x6ad6f90)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1986475&tab=buildLog) on [master@69bc0192b32d8204d0d980b0f7e9c2650a9383cd](https://github.com/cockroachdb/cockroach/commits/69bc0192b32d8204d0d980b0f7e9c2650a9383cd):

```
W200603 06:25:11.234199 517796 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200603 06:25:11.230264 448325 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 2: EOF:
W200603 06:25:11.242349 446688 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 2: EOF:
I200603 06:25:11.248446 441454 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200603 06:25:11.249038 441454 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200603 06:25:11.249471 441454 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (split): node unavailable; try another peer
I200603 06:25:11.249917 441454 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200603 06:25:11.250140 441454 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200603 06:25:11.250339 441454 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200603 06:25:11.250561 441454 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200603 06:25:11.250868 441454 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200603 06:25:11.251271 441454 kv/kvserver/queue.go:579  [n2,s2] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200603 06:25:11.141456 832841 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 5: EOF:
W200603 06:25:11.259606 441971 gossip/gossip.go:1525  [n3] no incoming or outgoing connections
W200603 06:25:11.260557 517191 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200603 06:25:11.262797 447358 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 4: rpc error: code = Unavailable desc = transport is closing:
I200603 06:25:11.267813 1114924 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39791: context canceled
I200603 06:25:11.270752 1114921 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43313: context canceled
W200603 06:25:11.281322 446344 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: EOF:
I200603 06:25:11.282831 1114922 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36909: context canceled
W200603 06:25:11.284495 445479 kv/kvserver/raft_transport.go:637  [n4] while processing outgoing Raft queue to node 3: EOF:
I200603 06:25:11.288495 1114923 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:36263: context canceled
I200603 06:25:11.289577 1114822 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:36263: context canceled
I200603 06:25:11.310835 1114824 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43313: context canceled
I200603 06:25:11.313524 1114825 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36909: context canceled
I200603 06:25:11.314558 1114823 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:44049: context canceled
W200603 06:25:11.319691 445407 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Unavailable desc = transport is closing:
I200603 06:25:11.330100 1114947 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39791: context canceled
W200603 06:25:11.334546 671917 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200603 06:25:11.336567 443462 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200603 06:25:11.337216 443462 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200603 06:25:11.337681 443462 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (split): node unavailable; try another peer
I200603 06:25:11.338843 443462 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200603 06:25:11.339340 443462 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200603 06:25:11.339613 443462 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200603 06:25:11.344502 443462 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200603 06:25:11.345038 443462 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200603 06:25:11.347527 443462 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200603 06:25:11.375757 1114940 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39791: context canceled
I200603 06:25:11.377691 1114941 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:36263: context canceled
I200603 06:25:11.379769 1114939 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36909: context canceled
I200603 06:25:11.383811 1114942 rpc/nodedialer/nodedialer.go:160  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:44049: context canceled
W200603 06:25:11.414614 443445 kv/kvserver/store.go:1652  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200603 06:25:11.467126 443445 kv/kvserver/store.go:1652  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (411.10s)
    testing.go:853: race detected during execution of test


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateQueueDownReplicate failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1987173&tab=buildLog) on [master@69bc0192b32d8204d0d980b0f7e9c2650a9383cd](https://github.com/cockroachdb/cockroach/commits/69bc0192b32d8204d0d980b0f7e9c2650a9383cd):

```
W200603 15:28:50.016430 462855 kv/kvserver/store.go:1652  [n5,s5,r2/2:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200603 15:28:50.033043 462854 kv/kvserver/store.go:1652  [n5,s5,r6/2:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
I200603 15:28:50.061905 462830 gossip/gossip.go:577  [n5] gossip status (stalled, 5 nodes)
gossip client (0/3 cur/max conns)
gossip server (0/3 cur/max conns, infos 700/2186 sent/received, bytes 144919B/447655B sent/received)
gossip connectivity
W200603 15:28:50.064045 459209 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200603 15:28:50.066124 462855 kv/kvserver/store.go:1652  [n5,s5,r2/2:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200603 15:28:50.080748 462854 kv/kvserver/store.go:1652  [n5,s5,r6/2:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200603 15:28:50.154367 462855 kv/kvserver/store.go:1652  [n5,s5,r2/2:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200603 15:28:50.170164 462854 kv/kvserver/store.go:1652  [n5,s5,r6/2:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
I200603 15:28:50.256470 1120775 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] storage.Replica: computing checksum
W200603 15:28:50.323143 460628 kv/kvserver/store.go:1652  [n4,s4,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200603 15:28:50.358202 462854 kv/kvserver/store.go:1652  [n5,s5,r6/2:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200603 15:28:50.366585 462855 kv/kvserver/store.go:1652  [n5,s5,r2/2:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200603 15:28:50.445456 458490 kv/kvserver/store.go:1652  [n2,s2,r2/5:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200603 15:28:50.510251 463476 server/node.go:718  [n5,summaries] error recording status summaries: node unavailable; try another peer
I200603 15:28:50.513481 459228 kv/kvserver/queue.go:579  [n3,s3] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200603 15:28:50.546305 462879 server/status/runtime.go:499  [n5] runtime stats: 12 GiB RSS, 650 goroutines, 1.5 GiB/105 MiB/1.7 GiB GO alloc/idle/total(stale), 179 MiB/236 MiB CGO alloc/total, 2006.1 CGO/sec, 302.0/29.5 %(u/s)time, 0.0 %gc (0x), 1.1 MiB/1.1 MiB (r/w)net
W200603 15:28:50.619668 459211 kv/kvserver/store.go:1652  [n3,s3,r2/4:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200603 15:28:50.683118 456236 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200603 15:28:50.718225 459210 kv/kvserver/store.go:1652  [n3,s3,r6/4:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200603 15:28:50.753368 462854 kv/kvserver/store.go:1652  [n5,s5,r6/2:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200603 15:28:50.753742 458488 kv/kvserver/store.go:1652  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200603 15:28:50.806836 460629 kv/kvserver/store.go:1652  [n4,s4,r6/3:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200603 15:28:50.817668 462855 kv/kvserver/store.go:1652  [n5,s5,r2/2:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200603 15:28:50.818922 462853 kv/kvserver/store.go:1652  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200603 15:28:50.823713 456263 kv/kvserver/queue.go:579  [n1,s1] rate limited in MaybeAdd (merge): node unavailable; try another peer
W200603 15:28:50.830063 460630 kv/kvserver/store.go:1652  [n4,s4,r2/3:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200603 15:28:50.974692 458489 kv/kvserver/store.go:1652  [n2,s2,r6/5:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200603 15:28:51.025233 459209 kv/kvserver/store.go:1652  [n3,s3,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200603 15:28:51.272671 460652 kv/kvserver/queue.go:579  [n4,s4] rate limited in MaybeAdd (merge): node unavailable; try another peer
--- FAIL: TestReplicateQueueDownReplicate (427.68s)
    replicate_queue_test.go:287: condition failed to evaluate within 45s: AdminChangeReplicas error: change replicas of r33 failed: descriptor changed: [expected] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):4, (n5,s5):3, (n2,s2):8LEARNER, next=9, gen=20, sticky=9223372036.854775807,2147483647] != [actual] r33:{m-/Table/SystemConfigSpan/Start} [(n1,s1):1, (n4,s4):4, (n5,s5):3, next=9, gen=21, sticky=9223372036.854775807,2147483647]
        goroutine 456076 [running]:
        runtime/debug.Stack(0xc0196bb9a0, 0x7a6fb40, 0xc04a410e60)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7ba6400, 0xc049718e00, 0xc0196bb9a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueDownReplicate(0xc049718e00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:287 +0x694
        testing.tRunner(0xc049718e00, 0x6aeaaf8)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateQueueDownReplicate PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateQueueDownReplicate.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -241,6 +241,7 @@ func TestReplicateQueueUpReplicate(t *testing.T) {
 // notice over-replicated ranges and remove replicas from them.
 func TestReplicateQueueDownReplicate(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
 	const replicaCount = 3
 
 	// The goal of this test is to ensure that down replication occurs correctly
@@ -255,46 +256,22 @@ func TestReplicateQueueDownReplicate(t *testing.T) {
 			},
 		},
 	)
-	defer tc.Stopper().Stop(context.Background())
+	defer tc.Stopper().Stop(ctx)
 
-	// Split off a range from the initial range for testing; there are
-	// complications if the metadata ranges are moved.
-	testKey := roachpb.Key("m")
-	if _, _, err := tc.SplitRange(testKey); err != nil {
-		t.Fatal(err)
-	}
+	// Disable the replication queues so that the range we're about to create
+	// doesn't get down-replicated too soon.
+	tc.ToggleReplicateQueues(false)
 
-	allowedErrs := strings.Join([]string{
-		// If a node is already present, we expect this error.
-		"unable to add replica .* which is already present",
-		// If a replica for this range was previously present on this store and
-		// it has already been removed but has not yet been GCed, this error
-		// is expected.
-		kvserver.IntersectingSnapshotMsg,
-	}, "|")
-
-	// Up-replicate the new range to all nodes to create redundant replicas.
-	// Every time a new replica is added, there's a very good chance that
-	// another one is removed. So all the replicas can't be added at once and
-	// instead need to be added one at a time ensuring that the replica did
-	// indeed make it to the desired target.
-	for _, server := range tc.Servers {
-		nodeID := server.NodeID()
-		// If this is not wrapped in a SucceedsSoon, then other temporary
-		// failures unlike the ones listed below, such as rejected reservations
-		// can cause the test to fail. When encountering those failures, a
-		// retry is in order.
-		testutils.SucceedsSoon(t, func() error {
-			_, err := tc.AddReplicas(testKey, roachpb.ReplicationTarget{
-				NodeID:  nodeID,
-				StoreID: server.GetFirstStoreID(),
-			})
-			if testutils.IsError(err, allowedErrs) {
-				return nil
-			}
-			return err
-		})
-	}
+	testKey := tc.ScratchRange(t)
+	desc := tc.LookupRangeOrFatal(t, testKey)
+	// At the end of StartTestCluster(), all ranges have 5 replicas since they're
+	// all "system ranges". When the ScratchRange() splits its range, it also
+	// starts up with 5 replicas. Since it's not a system range, its default zone
+	// config asks for 3x replication, and the replication queue will
+	// down-replicate it.
+	require.Len(t, desc.Replicas().All(), 5)
+	// Re-enable the replication queue.
+	tc.ToggleReplicateQueues(true)
 
 	// Now wait until the replicas have been down-replicated back to the
 	// desired number.

--- a/pkg/testutils/testcluster/testcluster.go
+++ b/pkg/testutils/testcluster/testcluster.go
@@ -753,6 +753,7 @@ func (tc *TestCluster) findMemberStore(storeID roachpb.StoreID) (*kvserver.Store
 // TODO(andrei): This method takes inexplicably long.
 // I think it shouldn't need any retries. See #38565.
 func (tc *TestCluster) WaitForFullReplication() error {
+	log.Infof(context.TODO(), "WaitForFullReplication")
 	start := timeutil.Now()
 	defer func() {
 		end := timeutil.Now()
@@ -872,6 +873,17 @@ func (tc *TestCluster) WaitForNodeLiveness(t testing.TB) {
 // ReplicationMode implements TestClusterInterface.
 func (tc *TestCluster) ReplicationMode() base.TestClusterReplicationMode {
 	return tc.replicationMode
+}
+
+// ToggleReplicateQueues activates or deactivates the replication queues on all
+// the stores on all the nodes.
+func (tc *TestCluster) ToggleReplicateQueues(active bool) {
+	for _, s := range tc.Servers {
+		_ = s.Stores().VisitStores(func(store *kvserver.Store) error {
+			store.SetReplicateQueueActive(active)
+			return nil
+		})
+	}
 }
 
 type testClusterFactoryImpl struct{}

